---
title: "10X-scData_3-Visualization-calculation-DVGs"
author: "Chang Wang"
date: "7/31/2018"
output: 
  html_document: default
  pdf_document: default
---
```{r setup, include=FALSE}
require("knitr")
opts_knit$set(root.dir="~/Documents/10XGenomics_BulkRNAseq_qPCR_2017Fall/Writing_summarization/Codes/R/")
```

###### Visualize of defective viral transcripts (DIP) and calculate their relative abundance ######
```{r warning=FALSE, message=FALSE}
library(ggplot2)
library(gplots)
library(RColorBrewer)
library(reshape)
library(ggthemes)
library(scales)
library(plyr)
library(dplyr)
library(Hmisc)

"%||%" <- function(a, b) {
  if (!is.null(a)) a else b
}

geom_flat_violin <- function(mapping = NULL, data = NULL, stat = "ydensity",
                             position = "dodge", trim = TRUE, scale = "area",
                             show.legend = NA, inherit.aes = TRUE, ...) {
  layer(
    data = data,
    mapping = mapping,
    stat = stat,
    geom = GeomFlatViolin,
    position = position,
    show.legend = show.legend,
    inherit.aes = inherit.aes,
    params = list(
      trim = trim,
      scale = scale,
      ...
    )
  )
}

GeomFlatViolin <-
  ggproto("GeomFlatViolin", Geom,
          setup_data = function(data, params) {
            data$width <- data$width %||%
              params$width %||% (resolution(data$x, FALSE) * 0.9)
            # ymin, ymax, xmin, and xmax define the bounding rectangle for each group
            data %>%
              group_by(group) %>%
              mutate(ymin = min(y),
                     ymax = max(y),
                     xmin = x,
                     xmax = x + width / 2) 
          },
          draw_group = function(data, panel_scales, coord) {
            # Find the points for the line to go all the way around
            data <- transform(data, xminv = x,
                              xmaxv = x + violinwidth * (xmax - x))
            # Make sure it's sorted properly to draw the outline
            newdata <- rbind(plyr::arrange(transform(data, x = xminv), y),
                             plyr::arrange(transform(data, x = xmaxv), -y))
            # Close the polygon: set first and last point the same
            # Needed for coord_polar and such
            newdata <- rbind(newdata, newdata[1,])
            ggplot2:::ggname("geom_flat_violin", 
                             GeomPolygon$draw_panel(newdata, panel_scales, coord)) 
          },
          draw_key = draw_key_polygon,
          default_aes = aes(weight = 1, colour = "grey20", fill = "white", size = 0.5,
                            alpha = NA, linetype = "solid"),
          required_aes = c("x", "y")
  )

```



#### Load all the DIP boundaries for scRNA-seq data, bulk RNA-seq data, and stock sequencing data ####
```{r warning=FALSE, message=FALSE}
setwd("~/Documents/10XGenomics_BulkRNAseq_qPCR_2017Fall/10X_data_July2018_v2/")

## PR8 stock ##
stock_PB2_boundary5 <- read.csv("PR8_virus_stock/DIP_stats/AF389115.1_boundary_5prime-end.csv",
                                row.names = 1)
stock_PB2_boundary3 <- read.csv("PR8_virus_stock/DIP_stats/AF389115.1_boundary_3prime-end.csv",
                                row.names = 1)
stock_PB1_boundary5 <- read.csv("PR8_virus_stock/DIP_stats/AF389116.1_boundary_5prime-end.csv",
                                row.names = 1)
stock_PB1_boundary3 <- read.csv("PR8_virus_stock/DIP_stats/AF389116.1_boundary_3prime-end.csv",
                                row.names = 1)
stock_PA_boundary5 <- read.csv("PR8_virus_stock/DIP_stats/AF389117.1_boundary_5prime-end.csv",
                               row.names = 1)
stock_PA_boundary3 <- read.csv("PR8_virus_stock/DIP_stats/AF389117.1_boundary_3prime-end.csv",
                               row.names = 1)

## scRNA-seq data ##
A549_6h_PB2_boundary5 <- read.csv("A549_6h/DIP_UMI_stats/AF389115.1_boundary_5prime-end.csv",
                                  row.names = 1)
A549_6h_PB2_boundary3 <- read.csv("A549_6h/DIP_UMI_stats/AF389115.1_boundary_3prime-end.csv",
                                  row.names = 1)
A549_6h_PB1_boundary5 <- read.csv("A549_6h/DIP_UMI_stats/AF389116.1_boundary_5prime-end.csv",
                                  row.names = 1)
A549_6h_PB1_boundary3 <- read.csv("A549_6h/DIP_UMI_stats/AF389116.1_boundary_3prime-end.csv",
                                  row.names = 1)
A549_6h_PA_boundary5 <- read.csv("A549_6h/DIP_UMI_stats/AF389117.1_boundary_5prime-end.csv",
                                 row.names = 1)
A549_6h_PA_boundary3 <- read.csv("A549_6h/DIP_UMI_stats/AF389117.1_boundary_3prime-end.csv",
                                 row.names = 1)

A549_12h_PB2_boundary5 <- read.csv("A549_12h/DIP_UMI_stats/AF389115.1_boundary_5prime-end.csv",
                                   row.names = 1)
A549_12h_PB2_boundary3 <- read.csv("A549_12h/DIP_UMI_stats/AF389115.1_boundary_3prime-end.csv",
                                   row.names = 1)
A549_12h_PB1_boundary5 <- read.csv("A549_12h/DIP_UMI_stats/AF389116.1_boundary_5prime-end.csv",
                                   row.names = 1)
A549_12h_PB1_boundary3 <- read.csv("A549_12h/DIP_UMI_stats/AF389116.1_boundary_3prime-end.csv",
                                   row.names = 1)
A549_12h_PA_boundary5 <- read.csv("A549_12h/DIP_UMI_stats/AF389117.1_boundary_5prime-end.csv",
                                  row.names = 1)
A549_12h_PA_boundary3 <- read.csv("A549_12h/DIP_UMI_stats/AF389117.1_boundary_3prime-end.csv",
                                  row.names = 1)

A549_24h_PB2_boundary5 <- read.csv("A549_24h/DIP_UMI_stats/AF389115.1_boundary_5prime-end.csv",
                                   row.names = 1)
A549_24h_PB2_boundary3 <- read.csv("A549_24h/DIP_UMI_stats/AF389115.1_boundary_3prime-end.csv",
                                   row.names = 1)
A549_24h_PB1_boundary5 <- read.csv("A549_24h/DIP_UMI_stats/AF389116.1_boundary_5prime-end.csv",
                                   row.names = 1)
A549_24h_PB1_boundary3 <- read.csv("A549_24h/DIP_UMI_stats/AF389116.1_boundary_3prime-end.csv",
                                   row.names = 1)
A549_24h_PA_boundary5 <- read.csv("A549_24h/DIP_UMI_stats/AF389117.1_boundary_5prime-end.csv",
                                  row.names = 1)
A549_24h_PA_boundary3 <- read.csv("A549_24h/DIP_UMI_stats/AF389117.1_boundary_3prime-end.csv",
                                  row.names = 1)

HBEpC_6h_PB2_boundary5 <- read.csv("HBEpC_6h/DIP_UMI_stats/AF389115.1_boundary_5prime-end.csv",
                                   row.names = 1, stringsAsFactors=FALSE)
HBEpC_6h_PB2_boundary3 <- read.csv("HBEpC_6h/DIP_UMI_stats/AF389115.1_boundary_3prime-end.csv",
                                   row.names = 1, stringsAsFactors=FALSE)
HBEpC_6h_PB1_boundary5 <- read.csv("HBEpC_6h/DIP_UMI_stats/AF389116.1_boundary_5prime-end.csv",
                                   row.names = 1, stringsAsFactors=FALSE)
HBEpC_6h_PB1_boundary3 <- read.csv("HBEpC_6h/DIP_UMI_stats/AF389116.1_boundary_3prime-end.csv",
                                   row.names = 1, stringsAsFactors=FALSE)
HBEpC_6h_PA_boundary5 <- read.csv("HBEpC_6h/DIP_UMI_stats/AF389117.1_boundary_5prime-end.csv",
                                  row.names = 1, stringsAsFactors=FALSE)
HBEpC_6h_PA_boundary3 <- read.csv("HBEpC_6h/DIP_UMI_stats/AF389117.1_boundary_3prime-end.csv",
                                  row.names = 1, stringsAsFactors=FALSE)

HBEpC_12h_PB2_boundary5 <- read.csv("HBEpC_12h/DIP_UMI_stats/AF389115.1_boundary_5prime-end.csv",
                                    row.names = 1)
HBEpC_12h_PB2_boundary3 <- read.csv("HBEpC_12h/DIP_UMI_stats/AF389115.1_boundary_3prime-end.csv",
                                    row.names = 1)
HBEpC_12h_PB1_boundary5 <- read.csv("HBEpC_12h/DIP_UMI_stats/AF389116.1_boundary_5prime-end.csv",
                                    row.names = 1)
HBEpC_12h_PB1_boundary3 <- read.csv("HBEpC_12h/DIP_UMI_stats/AF389116.1_boundary_3prime-end.csv",
                                    row.names = 1)
HBEpC_12h_PA_boundary5 <- read.csv("HBEpC_12h/DIP_UMI_stats/AF389117.1_boundary_5prime-end.csv",
                                   row.names = 1)
HBEpC_12h_PA_boundary3 <- read.csv("HBEpC_12h/DIP_UMI_stats/AF389117.1_boundary_3prime-end.csv",
                                   row.names = 1)

HBEpC_24h_PB2_boundary5 <- read.csv("HBEpC_24h/DIP_UMI_stats/AF389115.1_boundary_5prime-end.csv",
                                    row.names = 1)
HBEpC_24h_PB2_boundary3 <- read.csv("HBEpC_24h/DIP_UMI_stats/AF389115.1_boundary_3prime-end.csv",
                                    row.names = 1)
HBEpC_24h_PB1_boundary5 <- read.csv("HBEpC_24h/DIP_UMI_stats/AF389116.1_boundary_5prime-end.csv",
                                    row.names = 1)
HBEpC_24h_PB1_boundary3 <- read.csv("HBEpC_24h/DIP_UMI_stats/AF389116.1_boundary_3prime-end.csv",
                                    row.names = 1)
HBEpC_24h_PA_boundary5 <- read.csv("HBEpC_24h/DIP_UMI_stats/AF389117.1_boundary_5prime-end.csv",
                                   row.names = 1)
HBEpC_24h_PA_boundary3 <- read.csv("HBEpC_24h/DIP_UMI_stats/AF389117.1_boundary_3prime-end.csv",
                                   row.names = 1)


## Load the info for QCed cell ids for single cell data ##
load("CellQC/A549/A549ALLsamples_QCed_gene-expression-matrix_S2_A549-6h.RData")
load("CellQC/A549/A549ALLsamples_QCed_gene-expression-matrix_S3_A549-12h.RData")
load("CellQC/A549/A549ALLsamples_QCed_gene-expression-matrix_S4_A549-24h.RData")
load("CellQC/HBEpC/HBEpCALLsamples_QCed_gene-expression-matrix_s6_HBEpC-6h.RData")
load("CellQC/HBEpC/HBEpCALLsamples_QCed_gene-expression-matrix_s7_HBEpC-12h.RData")
load("CellQC/HBEpC/HBEpCALLsamples_QCed_gene-expression-matrix_s8_HBEpC-24h.RData")


## Check cells full of na in the sc data ##
sum(A549_6h_PB2_boundary5$X0 == "na")
sum(A549_6h_PB2_boundary3$X0 == "na")
sum(A549_6h_PB1_boundary5$X0 == "na")
sum(A549_6h_PB1_boundary3$X0 == "na")
sum(A549_6h_PA_boundary5$X0 == "na") 
sum(A549_6h_PA_boundary3$X0 == "na") 

sum(A549_12h_PB2_boundary5$X0 == "na")
sum(A549_12h_PB2_boundary3$X0 == "na")
sum(A549_12h_PB1_boundary5$X0 == "na")
sum(A549_12h_PB1_boundary3$X0 == "na")
sum(A549_12h_PA_boundary5$X0 == "na") 
sum(A549_12h_PA_boundary3$X0 == "na") 

sum(A549_24h_PB2_boundary5$X0 == "na")
sum(A549_24h_PB2_boundary3$X0 == "na")
sum(A549_24h_PB1_boundary5$X0 == "na")
sum(A549_24h_PB1_boundary3$X0 == "na")
sum(A549_24h_PA_boundary5$X0 == "na") 
sum(A549_24h_PA_boundary3$X0 == "na") 

sum(HBEpC_6h_PB2_boundary5$X0 == "na")
sum(HBEpC_6h_PB2_boundary3$X0 == "na")
sum(HBEpC_6h_PB1_boundary5$X0 == "na")
sum(HBEpC_6h_PB1_boundary3$X0 == "na")
sum(HBEpC_6h_PA_boundary5$X0 == "na") 
sum(HBEpC_6h_PA_boundary3$X0 == "na") 

sum(HBEpC_12h_PB2_boundary5$X0 == "na")
sum(HBEpC_12h_PB2_boundary3$X0 == "na")
sum(HBEpC_12h_PB1_boundary5$X0 == "na")
sum(HBEpC_12h_PB1_boundary3$X0 == "na")
sum(HBEpC_12h_PA_boundary5$X0 == "na") 
sum(HBEpC_12h_PA_boundary3$X0 == "na") 

sum(HBEpC_24h_PB2_boundary5$X0 == "na")
sum(HBEpC_24h_PB2_boundary3$X0 == "na")
sum(HBEpC_24h_PB1_boundary5$X0 == "na")
sum(HBEpC_24h_PB1_boundary3$X0 == "na")
sum(HBEpC_24h_PA_boundary5$X0 == "na") 
sum(HBEpC_24h_PA_boundary3$X0 == "na") 

```



#### Remove cells with no boundaries detected from each sc sample ####
```{r warning=FALSE, message=FALSE}
A549_6h_PB2_boundary5 <- A549_6h_PB2_boundary5[A549_6h_PB2_boundary5$X0 != "na", ]
A549_6h_PB2_boundary3 <- A549_6h_PB2_boundary3[A549_6h_PB2_boundary3$X0 != "na", ]
A549_6h_PB1_boundary5 <- A549_6h_PB1_boundary5[A549_6h_PB1_boundary5$X0 != "na", ]
A549_6h_PB1_boundary3 <- A549_6h_PB1_boundary3[A549_6h_PB1_boundary3$X0 != "na", ]
A549_6h_PA_boundary5 <- A549_6h_PA_boundary5[A549_6h_PA_boundary5$X0 != "na", ]
A549_6h_PA_boundary3 <- A549_6h_PA_boundary3[A549_6h_PA_boundary3$X0 != "na", ]

A549_12h_PB2_boundary5 <- A549_12h_PB2_boundary5[A549_12h_PB2_boundary5$X0 != "na", ]
A549_12h_PB2_boundary3 <- A549_12h_PB2_boundary3[A549_12h_PB2_boundary3$X0 != "na", ]
A549_12h_PB1_boundary5 <- A549_12h_PB1_boundary5[A549_12h_PB1_boundary5$X0 != "na", ]
A549_12h_PB1_boundary3 <- A549_12h_PB1_boundary3[A549_12h_PB1_boundary3$X0 != "na", ]
A549_12h_PA_boundary5 <- A549_12h_PA_boundary5[A549_12h_PA_boundary5$X0 != "na", ]
A549_12h_PA_boundary3 <- A549_12h_PA_boundary3[A549_12h_PA_boundary3$X0 != "na", ]

A549_24h_PB2_boundary5 <- A549_24h_PB2_boundary5[A549_24h_PB2_boundary5$X0 != "na", ]
A549_24h_PB2_boundary3 <- A549_24h_PB2_boundary3[A549_24h_PB2_boundary3$X0 != "na", ]
A549_24h_PB1_boundary5 <- A549_24h_PB1_boundary5[A549_24h_PB1_boundary5$X0 != "na", ]
A549_24h_PB1_boundary3 <- A549_24h_PB1_boundary3[A549_24h_PB1_boundary3$X0 != "na", ]
A549_24h_PA_boundary5 <- A549_24h_PA_boundary5[A549_24h_PA_boundary5$X0 != "na", ]
A549_24h_PA_boundary3 <- A549_24h_PA_boundary3[A549_24h_PA_boundary3$X0 != "na", ]

HBEpC_6h_PB2_boundary5 <- HBEpC_6h_PB2_boundary5[HBEpC_6h_PB2_boundary5$X0 != "na", ]
HBEpC_6h_PB2_boundary3 <- HBEpC_6h_PB2_boundary3[HBEpC_6h_PB2_boundary3$X0 != "na", ]
HBEpC_6h_PB1_boundary5 <- HBEpC_6h_PB1_boundary5[HBEpC_6h_PB1_boundary5$X0 != "na", ]
HBEpC_6h_PB1_boundary3 <- HBEpC_6h_PB1_boundary3[HBEpC_6h_PB1_boundary3$X0 != "na", ]
HBEpC_6h_PA_boundary5 <- HBEpC_6h_PA_boundary5[HBEpC_6h_PA_boundary5$X0 != "na", ]
HBEpC_6h_PA_boundary3 <- HBEpC_6h_PA_boundary3[HBEpC_6h_PA_boundary3$X0 != "na", ]

HBEpC_12h_PB2_boundary5 <- HBEpC_12h_PB2_boundary5[HBEpC_12h_PB2_boundary5$X0 != "na", ]
HBEpC_12h_PB2_boundary3 <- HBEpC_12h_PB2_boundary3[HBEpC_12h_PB2_boundary3$X0 != "na", ]
HBEpC_12h_PB1_boundary5 <- HBEpC_12h_PB1_boundary5[HBEpC_12h_PB1_boundary5$X0 != "na", ]
HBEpC_12h_PB1_boundary3 <- HBEpC_12h_PB1_boundary3[HBEpC_12h_PB1_boundary3$X0 != "na", ]
HBEpC_12h_PA_boundary5 <- HBEpC_12h_PA_boundary5[HBEpC_12h_PA_boundary5$X0 != "na", ]
HBEpC_12h_PA_boundary3 <- HBEpC_12h_PA_boundary3[HBEpC_12h_PA_boundary3$X0 != "na", ]

HBEpC_24h_PB2_boundary5 <- HBEpC_24h_PB2_boundary5[HBEpC_24h_PB2_boundary5$X0 != "na", ]
HBEpC_24h_PB2_boundary3 <- HBEpC_24h_PB2_boundary3[HBEpC_24h_PB2_boundary3$X0 != "na", ]
HBEpC_24h_PB1_boundary5 <- HBEpC_24h_PB1_boundary5[HBEpC_24h_PB1_boundary5$X0 != "na", ]
HBEpC_24h_PB1_boundary3 <- HBEpC_24h_PB1_boundary3[HBEpC_24h_PB1_boundary3$X0 != "na", ]
HBEpC_24h_PA_boundary5 <- HBEpC_24h_PA_boundary5[HBEpC_24h_PA_boundary5$X0 != "na", ]
HBEpC_24h_PA_boundary3 <- HBEpC_24h_PA_boundary3[HBEpC_24h_PA_boundary3$X0 != "na", ]

```



#### Get the paired boundaries (5' and 3') for three polymerases in PR8 stock ####
```{r warning=FALSE, message=FALSE, include=TRUE, eval=FALSE}
stock_PB2_boundary <- data.frame(cbind(t(stock_PB2_boundary5), t(stock_PB2_boundary3)))
colnames(stock_PB2_boundary) <- c("start", "end")
stock_PB2_boundary$sample <- rep("PR8 stock", nrow(stock_PB2_boundary))
dim(unique(stock_PB2_boundary)) 

stock_PB1_boundary <- data.frame(cbind(t(stock_PB1_boundary5), t(stock_PB1_boundary3)))
colnames(stock_PB1_boundary) <- c("start", "end")
stock_PB1_boundary$sample <- rep("PR8 stock", nrow(stock_PB1_boundary))
dim(unique(stock_PB1_boundary)) 

stock_PA_boundary <- data.frame(cbind(t(stock_PA_boundary5), t(stock_PA_boundary3)))
colnames(stock_PA_boundary) <- c("start", "end")
stock_PA_boundary$sample <- rep("PR8 stock", nrow(stock_PA_boundary))
dim(unique(stock_PA_boundary)) 

```



#### Get the paried boundaries (5' and 3') for three polymerases in scRNA-seq data ####
```{r warning=FALSE, message=FALSE, include=TRUE, eval=FALSE}
## A549 6h ##
# PB2
A549_6h_PB2_boundary3 <- A549_6h_PB2_boundary3[rownames(A549_6h_PB2_boundary5), ]
A549_6h_sc_PB2_boundary <- list()
for(i in 1:nrow(A549_6h_PB2_boundary5)) {
  boundary5 <- data.frame(t(A549_6h_PB2_boundary5[i, A549_6h_PB2_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(A549_6h_PB2_boundary3[i, A549_6h_PB2_boundary3[i, ] != "na"]))
  A549_6h_sc_PB2_boundary[[i]] <- data.frame("start"=boundary5, 
                                             "stop"=boundary3,
                                             "cell"=rownames(A549_6h_PB2_boundary5)[i])
  colnames(A549_6h_sc_PB2_boundary[[i]])[1:2] <- c("start", "stop")
}
A549_6h_sc_PB2_boundary_df <- do.call("rbind", A549_6h_sc_PB2_boundary)
A549_6h_sc_PB2_boundary_df$start <- as.numeric(as.character(A549_6h_sc_PB2_boundary_df$start))
A549_6h_sc_PB2_boundary_df$stop <- as.numeric(as.character(A549_6h_sc_PB2_boundary_df$stop))
save(A549_6h_sc_PB2_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_A549_6h_PB2_boundaries.RData")
dim(unique(A549_6h_sc_PB2_boundary_df[,1:2])) 

# PB1
A549_6h_PB1_boundary3 <- A549_6h_PB1_boundary3[rownames(A549_6h_PB1_boundary5), ]
A549_6h_sc_PB1_boundary <- list()
for(i in 1:nrow(A549_6h_PB1_boundary5)) {
  boundary5 <- data.frame(t(A549_6h_PB1_boundary5[i, A549_6h_PB1_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(A549_6h_PB1_boundary3[i, A549_6h_PB1_boundary3[i, ] != "na"]))
  A549_6h_sc_PB1_boundary[[i]] <- data.frame("start"=boundary5, 
                                             "stop"=boundary3,
                                             "cell"=rownames(A549_6h_PB1_boundary5)[i])
  colnames(A549_6h_sc_PB1_boundary[[i]])[1:2] <- c("start", "stop")
}
A549_6h_sc_PB1_boundary_df <- do.call("rbind", A549_6h_sc_PB1_boundary)
A549_6h_sc_PB1_boundary_df$start <- as.numeric(as.character(A549_6h_sc_PB1_boundary_df$start))
A549_6h_sc_PB1_boundary_df$stop <- as.numeric(as.character(A549_6h_sc_PB1_boundary_df$stop))
save(A549_6h_sc_PB1_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_A549_6h_PB1_boundaries.RData")
dim(unique(A549_6h_sc_PB1_boundary_df[,1:2])) 

# PA
A549_6h_PA_boundary3 <- A549_6h_PA_boundary3[rownames(A549_6h_PA_boundary5), ]
A549_6h_sc_PA_boundary <- list()
for(i in 1:nrow(A549_6h_PA_boundary5)) {
  boundary5 <- data.frame(t(A549_6h_PA_boundary5[i, A549_6h_PA_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(A549_6h_PA_boundary3[i, A549_6h_PA_boundary3[i, ] != "na"]))
  A549_6h_sc_PA_boundary[[i]] <- data.frame("start"=boundary5, 
                                            "stop"=boundary3,
                                            "cell"=rownames(A549_6h_PA_boundary5)[i])
  colnames(A549_6h_sc_PA_boundary[[i]])[1:2] <- c("start", "stop")
} 
# Because of a bug in rbind, we have to make the first dataframe have more than one row if any of the dataframe in the rest has more than one row
# Do this manually... Move a dataframe with more than one row to the first
A549_6h_sc_PA_boundary_1 <- A549_6h_sc_PA_boundary[[1]]
A549_6h_sc_PA_boundary[[1]] <- A549_6h_sc_PA_boundary[[3]]
A549_6h_sc_PA_boundary[[3]] <- A549_6h_sc_PA_boundary_1
A549_6h_sc_PA_boundary_df <- do.call("rbind", A549_6h_sc_PA_boundary)
A549_6h_sc_PA_boundary_df$start <- as.numeric(as.character(A549_6h_sc_PA_boundary_df$start))
A549_6h_sc_PA_boundary_df$stop <- as.numeric(as.character(A549_6h_sc_PA_boundary_df$stop))
save(A549_6h_sc_PA_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_A549_6h_PA_boundaries.RData")
dim(unique(A549_6h_sc_PA_boundary_df[,1:2])) 


## A549 12h ##
# PB2
A549_12h_PB2_boundary3 <- A549_12h_PB2_boundary3[rownames(A549_12h_PB2_boundary5), ]
A549_12h_sc_PB2_boundary <- list()
for(i in 1:nrow(A549_12h_PB2_boundary5)) {
  boundary5 <- data.frame(t(A549_12h_PB2_boundary5[i, A549_12h_PB2_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(A549_12h_PB2_boundary3[i, A549_12h_PB2_boundary3[i, ] != "na"]))
  A549_12h_sc_PB2_boundary[[i]] <- data.frame("start"=boundary5, 
                                              "stop"=boundary3,
                                              "cell"=rownames(A549_12h_PB2_boundary5)[i])
  colnames(A549_12h_sc_PB2_boundary[[i]])[1:2] <- c("start", "stop")
} 
A549_12h_sc_PB2_boundary_df <- do.call("rbind", A549_12h_sc_PB2_boundary)
A549_12h_sc_PB2_boundary_df$start <- as.numeric(as.character(A549_12h_sc_PB2_boundary_df$start))
A549_12h_sc_PB2_boundary_df$stop <- as.numeric(as.character(A549_12h_sc_PB2_boundary_df$stop))
save(A549_12h_sc_PB2_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_A549_12h_PB2_boundaries.RData")
dim(unique(A549_12h_sc_PB2_boundary_df[,1:2])) 

# PB1
A549_12h_PB1_boundary3 <- A549_12h_PB1_boundary3[rownames(A549_12h_PB1_boundary5), ]
A549_12h_sc_PB1_boundary <- list()
for(i in 1:nrow(A549_12h_PB1_boundary5)) {
  boundary5 <- data.frame(t(A549_12h_PB1_boundary5[i, A549_12h_PB1_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(A549_12h_PB1_boundary3[i, A549_12h_PB1_boundary3[i, ] != "na"]))
  A549_12h_sc_PB1_boundary[[i]] <- data.frame("start"=boundary5, 
                                              "stop"=boundary3,
                                              "cell"=rownames(A549_12h_PB1_boundary5)[i])
  colnames(A549_12h_sc_PB1_boundary[[i]])[1:2] <- c("start", "stop")
} 
A549_12h_sc_PB1_boundary_df <- do.call("rbind", A549_12h_sc_PB1_boundary)
A549_12h_sc_PB1_boundary_df$start <- as.numeric(as.character(A549_12h_sc_PB1_boundary_df$start))
A549_12h_sc_PB1_boundary_df$stop <- as.numeric(as.character(A549_12h_sc_PB1_boundary_df$stop))
save(A549_12h_sc_PB1_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_A549_12h_PB1_boundaries.RData")
dim(unique(A549_12h_sc_PB1_boundary_df[,1:2])) 

# PA
A549_12h_PA_boundary3 <- A549_12h_PA_boundary3[rownames(A549_12h_PA_boundary5), ]
A549_12h_sc_PA_boundary <- list()
for(i in 1:nrow(A549_12h_PA_boundary5)) {
  boundary5 <- data.frame(t(A549_12h_PA_boundary5[i, A549_12h_PA_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(A549_12h_PA_boundary3[i, A549_12h_PA_boundary3[i, ] != "na"]))
  A549_12h_sc_PA_boundary[[i]] <- data.frame("start"=boundary5, 
                                             "stop"=boundary3,
                                             "cell"=rownames(A549_12h_PA_boundary5)[i])
  colnames(A549_12h_sc_PA_boundary[[i]])[1:2] <- c("start", "stop")
} 
A549_12h_sc_PA_boundary_df <- do.call("rbind", A549_12h_sc_PA_boundary)
A549_12h_sc_PA_boundary_df$start <- as.numeric(as.character(A549_12h_sc_PA_boundary_df$start))
A549_12h_sc_PA_boundary_df$stop <- as.numeric(as.character(A549_12h_sc_PA_boundary_df$stop))
save(A549_12h_sc_PA_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_A549_12h_PA_boundaries.RData")
dim(unique(A549_12h_sc_PA_boundary_df[,1:2])) 


## A549 24h ##
# PB2
A549_24h_PB2_boundary3 <- A549_24h_PB2_boundary3[rownames(A549_24h_PB2_boundary5), ]
A549_24h_sc_PB2_boundary <- list()
for(i in 1:nrow(A549_24h_PB2_boundary5)) {
  boundary5 <- data.frame(t(A549_24h_PB2_boundary5[i, A549_24h_PB2_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(A549_24h_PB2_boundary3[i, A549_24h_PB2_boundary3[i, ] != "na"]))
  A549_24h_sc_PB2_boundary[[i]] <- data.frame("start"=boundary5, 
                                              "stop"=boundary3,
                                              "cell"=rownames(A549_24h_PB2_boundary5)[i])
  colnames(A549_24h_sc_PB2_boundary[[i]])[1:2] <- c("start", "stop")
}
A549_24h_sc_PB2_boundary_df <- do.call("rbind", A549_24h_sc_PB2_boundary)
A549_24h_sc_PB2_boundary_df$start <- as.numeric(as.character(A549_24h_sc_PB2_boundary_df$start))
A549_24h_sc_PB2_boundary_df$stop <- as.numeric(as.character(A549_24h_sc_PB2_boundary_df$stop))
save(A549_24h_sc_PB2_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_A549_24h_PB2_boundaries.RData")
dim(unique(A549_24h_sc_PB2_boundary_df[,1:2])) 

# PB1
A549_24h_PB1_boundary3 <- A549_24h_PB1_boundary3[rownames(A549_24h_PB1_boundary5), ]
A549_24h_sc_PB1_boundary <- list()
for(i in 1:nrow(A549_24h_PB1_boundary5)) {
  boundary5 <- data.frame(t(A549_24h_PB1_boundary5[i, A549_24h_PB1_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(A549_24h_PB1_boundary3[i, A549_24h_PB1_boundary3[i, ] != "na"]))
  A549_24h_sc_PB1_boundary[[i]] <- data.frame("start"=boundary5, 
                                              "stop"=boundary3,
                                              "cell"=rownames(A549_24h_PB1_boundary5)[i])
  colnames(A549_24h_sc_PB1_boundary[[i]])[1:2] <- c("start", "stop")
}
A549_24h_sc_PB1_boundary_df <- do.call("rbind", A549_24h_sc_PB1_boundary)
A549_24h_sc_PB1_boundary_df$start <- as.numeric(as.character(A549_24h_sc_PB1_boundary_df$start))
A549_24h_sc_PB1_boundary_df$stop <- as.numeric(as.character(A549_24h_sc_PB1_boundary_df$stop))
save(A549_24h_sc_PB1_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_A549_24h_PB1_boundaries.RData")
dim(unique(A549_24h_sc_PB1_boundary_df[,1:2])) 

# PA
A549_24h_PA_boundary3 <- A549_24h_PA_boundary3[rownames(A549_24h_PA_boundary5), ]
A549_24h_sc_PA_boundary <- list()
for(i in 1:nrow(A549_24h_PA_boundary5)) {
  boundary5 <- data.frame(t(A549_24h_PA_boundary5[i, A549_24h_PA_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(A549_24h_PA_boundary3[i, A549_24h_PA_boundary3[i, ] != "na"]))
  A549_24h_sc_PA_boundary[[i]] <- data.frame("start"=boundary5, 
                                             "stop"=boundary3,
                                             "cell"=rownames(A549_24h_PA_boundary5)[i])
  colnames(A549_24h_sc_PA_boundary[[i]])[1:2] <- c("start", "stop")
}
A549_24h_sc_PA_boundary_df <- do.call("rbind", A549_24h_sc_PA_boundary)
A549_24h_sc_PA_boundary_df$start <- as.numeric(as.character(A549_24h_sc_PA_boundary_df$start))
A549_24h_sc_PA_boundary_df$stop <- as.numeric(as.character(A549_24h_sc_PA_boundary_df$stop))
save(A549_24h_sc_PA_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_A549_24h_PA_boundaries.RData")
dim(unique(A549_24h_sc_PA_boundary_df[,1:2])) 


## HBEpC 6h ##
# PB2
HBEpC_6h_PB2_boundary3 <- HBEpC_6h_PB2_boundary3[rownames(HBEpC_6h_PB2_boundary5), ]
HBEpC_6h_sc_PB2_boundary <- list()
for(i in 1:nrow(HBEpC_6h_PB2_boundary5)) {
  boundary5 <- data.frame(t(HBEpC_6h_PB2_boundary5[i, HBEpC_6h_PB2_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(HBEpC_6h_PB2_boundary3[i, HBEpC_6h_PB2_boundary3[i, ] != "na"]))
  HBEpC_6h_sc_PB2_boundary[[i]] <- data.frame("start"=boundary5, 
                                              "stop"=boundary3,
                                              "cell"=rownames(HBEpC_6h_PB2_boundary5)[i])
  colnames(HBEpC_6h_sc_PB2_boundary[[i]])[1:2] <- c("start", "stop")
}
HBEpC_6h_sc_PB2_boundary_1 <- HBEpC_6h_sc_PB2_boundary[[1]]
HBEpC_6h_sc_PB2_boundary[[1]] <- HBEpC_6h_sc_PB2_boundary[[4]]
HBEpC_6h_sc_PB2_boundary[[4]] <- HBEpC_6h_sc_PB2_boundary_1
HBEpC_6h_sc_PB2_boundary_df <- do.call("rbind", HBEpC_6h_sc_PB2_boundary)
HBEpC_6h_sc_PB2_boundary_df$start <- as.numeric(as.character(HBEpC_6h_sc_PB2_boundary_df$start))
HBEpC_6h_sc_PB2_boundary_df$stop <- as.numeric(as.character(HBEpC_6h_sc_PB2_boundary_df$stop))
save(HBEpC_6h_sc_PB2_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_HBEpC_6h_PB2_boundaries.RData")
dim(unique(HBEpC_6h_sc_PB2_boundary_df[,1:2])) 

# PB1
HBEpC_6h_PB1_boundary3 <- HBEpC_6h_PB1_boundary3[rownames(HBEpC_6h_PB1_boundary5), ]
HBEpC_6h_sc_PB1_boundary <- list()
for(i in 1:nrow(HBEpC_6h_PB1_boundary5)) {
  boundary5 <- data.frame(t(HBEpC_6h_PB1_boundary5[i, HBEpC_6h_PB1_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(HBEpC_6h_PB1_boundary3[i, HBEpC_6h_PB1_boundary3[i, ] != "na"]))
  HBEpC_6h_sc_PB1_boundary[[i]] <- data.frame("start"=boundary5, 
                                              "stop"=boundary3,
                                              "cell"=rownames(HBEpC_6h_PB1_boundary5)[i])
  colnames(HBEpC_6h_sc_PB1_boundary[[i]])[1:2] <- c("start", "stop")
}
# Because of a bug in rbind, we have to make the first dataframe have more than one row if any of the dataframe in the rest has more than one row
# Do this manually... Move a dataframe with more than one row to the first
HBEpC_6h_sc_PB1_boundary_1 <- HBEpC_6h_sc_PB1_boundary[[1]]
HBEpC_6h_sc_PB1_boundary[[1]] <- HBEpC_6h_sc_PB1_boundary[[2]]
HBEpC_6h_sc_PB1_boundary[[2]] <- HBEpC_6h_sc_PB1_boundary_1
HBEpC_6h_sc_PB1_boundary_df <- do.call("rbind", HBEpC_6h_sc_PB1_boundary)
HBEpC_6h_sc_PB1_boundary_df$start <- as.numeric(as.character(HBEpC_6h_sc_PB1_boundary_df$start))
HBEpC_6h_sc_PB1_boundary_df$stop <- as.numeric(as.character(HBEpC_6h_sc_PB1_boundary_df$stop))
save(HBEpC_6h_sc_PB1_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_HBEpC_6h_PB1_boundaries.RData")
dim(unique(HBEpC_6h_sc_PB1_boundary_df[,1:2])) 

# PA
HBEpC_6h_PA_boundary3 <- HBEpC_6h_PA_boundary3[rownames(HBEpC_6h_PA_boundary5), ]
HBEpC_6h_sc_PA_boundary <- list()
for(i in 1:nrow(HBEpC_6h_PA_boundary5)) {
  boundary5 <- data.frame(t(HBEpC_6h_PA_boundary5[i, HBEpC_6h_PA_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(HBEpC_6h_PA_boundary3[i, HBEpC_6h_PA_boundary3[i, ] != "na"]))
  HBEpC_6h_sc_PA_boundary[[i]] <- data.frame("start"=boundary5, 
                                             "stop"=boundary3,
                                             "cell"=rownames(HBEpC_6h_PA_boundary5)[i])
  colnames(HBEpC_6h_sc_PA_boundary[[i]])[1:2] <- c("start", "stop")
}
HBEpC_6h_sc_PA_boundary_df <- do.call("rbind", HBEpC_6h_sc_PA_boundary)
HBEpC_6h_sc_PA_boundary_df$start <- as.numeric(as.character(HBEpC_6h_sc_PA_boundary_df$start))
HBEpC_6h_sc_PA_boundary_df$stop <- as.numeric(as.character(HBEpC_6h_sc_PA_boundary_df$stop))
save(HBEpC_6h_sc_PA_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_HBEpC_6h_PA_boundaries.RData")
dim(unique(HBEpC_6h_sc_PA_boundary_df[,1:2]))


## HBEpC 12h ##
# PB2
HBEpC_12h_PB2_boundary3 <- HBEpC_12h_PB2_boundary3[rownames(HBEpC_12h_PB2_boundary5), ]
HBEpC_12h_sc_PB2_boundary <- list()
for(i in 1:nrow(HBEpC_12h_PB2_boundary5)) {
  boundary5 <- data.frame(t(HBEpC_12h_PB2_boundary5[i, HBEpC_12h_PB2_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(HBEpC_12h_PB2_boundary3[i, HBEpC_12h_PB2_boundary3[i, ] != "na"]))
  HBEpC_12h_sc_PB2_boundary[[i]] <- data.frame("start"=boundary5, 
                                               "stop"=boundary3,
                                               "cell"=rownames(HBEpC_12h_PB2_boundary5)[i])
  colnames(HBEpC_12h_sc_PB2_boundary[[i]])[1:2] <- c("start", "stop")
}
HBEpC_12h_sc_PB2_boundary_df <- do.call("rbind", HBEpC_12h_sc_PB2_boundary)
HBEpC_12h_sc_PB2_boundary_df$start <- as.numeric(as.character(HBEpC_12h_sc_PB2_boundary_df$start))
HBEpC_12h_sc_PB2_boundary_df$stop <- as.numeric(as.character(HBEpC_12h_sc_PB2_boundary_df$stop))
save(HBEpC_12h_sc_PB2_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_HBEpC_12h_PB2_boundaries.RData")
dim(unique(HBEpC_12h_sc_PB2_boundary_df[,1:2])) 

# PB1
HBEpC_12h_PB1_boundary3 <- HBEpC_12h_PB1_boundary3[rownames(HBEpC_12h_PB1_boundary5), ]
HBEpC_12h_sc_PB1_boundary <- list()
for(i in 1:nrow(HBEpC_12h_PB1_boundary5)) {
  boundary5 <- data.frame(t(HBEpC_12h_PB1_boundary5[i, HBEpC_12h_PB1_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(HBEpC_12h_PB1_boundary3[i, HBEpC_12h_PB1_boundary3[i, ] != "na"]))
  HBEpC_12h_sc_PB1_boundary[[i]] <- data.frame("start"=boundary5, 
                                               "stop"=boundary3,
                                               "cell"=rownames(HBEpC_12h_PB1_boundary5)[i])
  colnames(HBEpC_12h_sc_PB1_boundary[[i]])[1:2] <- c("start", "stop")
}
HBEpC_12h_sc_PB1_boundary_df <- do.call("rbind", HBEpC_12h_sc_PB1_boundary)
HBEpC_12h_sc_PB1_boundary_df$start <- as.numeric(as.character(HBEpC_12h_sc_PB1_boundary_df$start))
HBEpC_12h_sc_PB1_boundary_df$stop <- as.numeric(as.character(HBEpC_12h_sc_PB1_boundary_df$stop))
save(HBEpC_12h_sc_PB1_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_HBEpC_12h_PB1_boundaries.RData")
dim(unique(HBEpC_12h_sc_PB1_boundary_df[,1:2])) 

# PA
HBEpC_12h_PA_boundary3 <- HBEpC_12h_PA_boundary3[rownames(HBEpC_12h_PA_boundary5), ]
HBEpC_12h_sc_PA_boundary <- list()
for(i in 1:nrow(HBEpC_12h_PA_boundary5)) {
  boundary5 <- data.frame(t(HBEpC_12h_PA_boundary5[i, HBEpC_12h_PA_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(HBEpC_12h_PA_boundary3[i, HBEpC_12h_PA_boundary3[i, ] != "na"]))
  HBEpC_12h_sc_PA_boundary[[i]] <- data.frame("start"=boundary5, 
                                              "stop"=boundary3,
                                              "cell"=rownames(HBEpC_12h_PA_boundary5)[i])
  colnames(HBEpC_12h_sc_PA_boundary[[i]])[1:2] <- c("start", "stop")
}
# Because of a bug in rbind, we have to make the first dataframe have more than one row if any of the dataframe in the rest has more than one row
# Do this manually... Move a dataframe with more than one row to the first
HBEpC_12h_sc_PA_boundary_1 <- HBEpC_12h_sc_PA_boundary[[1]]
HBEpC_12h_sc_PA_boundary[[1]] <- HBEpC_12h_sc_PA_boundary[[7]]
HBEpC_12h_sc_PA_boundary[[7]] <- HBEpC_12h_sc_PA_boundary_1
HBEpC_12h_sc_PA_boundary_df <- do.call("rbind", HBEpC_12h_sc_PA_boundary)
HBEpC_12h_sc_PA_boundary_df$start <- as.numeric(as.character(HBEpC_12h_sc_PA_boundary_df$start))
HBEpC_12h_sc_PA_boundary_df$stop <- as.numeric(as.character(HBEpC_12h_sc_PA_boundary_df$stop))
save(HBEpC_12h_sc_PA_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_HBEpC_12h_PA_boundaries.RData")
dim(unique(HBEpC_12h_sc_PA_boundary_df[,1:2])) 


## HBEpC 24h ##
# PB2
HBEpC_24h_PB2_boundary3 <- HBEpC_24h_PB2_boundary3[rownames(HBEpC_24h_PB2_boundary5), ]
HBEpC_24h_sc_PB2_boundary <- list()
for(i in 1:nrow(HBEpC_24h_PB2_boundary5)) {
  boundary5 <- data.frame(t(HBEpC_24h_PB2_boundary5[i, HBEpC_24h_PB2_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(HBEpC_24h_PB2_boundary3[i, HBEpC_24h_PB2_boundary3[i, ] != "na"]))
  HBEpC_24h_sc_PB2_boundary[[i]] <- data.frame("start"=boundary5, 
                                               "stop"=boundary3,
                                               "cell"=rownames(HBEpC_24h_PB2_boundary5)[i])
  colnames(HBEpC_24h_sc_PB2_boundary[[i]])[1:2] <- c("start", "stop")
}
HBEpC_24h_sc_PB2_boundary_df <- do.call("rbind", HBEpC_24h_sc_PB2_boundary)
HBEpC_24h_sc_PB2_boundary_df$start <- as.numeric(as.character(HBEpC_24h_sc_PB2_boundary_df$start))
HBEpC_24h_sc_PB2_boundary_df$stop <- as.numeric(as.character(HBEpC_24h_sc_PB2_boundary_df$stop))
save(HBEpC_24h_sc_PB2_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_HBEpC_24h_PB2_boundaries.RData")
dim(unique(HBEpC_24h_sc_PB2_boundary_df[,1:2])) 

# PB1
HBEpC_24h_PB1_boundary3 <- HBEpC_24h_PB1_boundary3[rownames(HBEpC_24h_PB1_boundary5), ]
HBEpC_24h_sc_PB1_boundary <- list()
for(i in 1:nrow(HBEpC_24h_PB1_boundary5)) {
  boundary5 <- data.frame(t(HBEpC_24h_PB1_boundary5[i, HBEpC_24h_PB1_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(HBEpC_24h_PB1_boundary3[i, HBEpC_24h_PB1_boundary3[i, ] != "na"]))
  HBEpC_24h_sc_PB1_boundary[[i]] <- data.frame("start"=boundary5, 
                                               "stop"=boundary3,
                                               "cell"=rownames(HBEpC_24h_PB1_boundary5)[i])
  colnames(HBEpC_24h_sc_PB1_boundary[[i]])[1:2] <- c("start", "stop")
}
HBEpC_24h_sc_PB1_boundary_df <- do.call("rbind", HBEpC_24h_sc_PB1_boundary)
HBEpC_24h_sc_PB1_boundary_df$start <- as.numeric(as.character(HBEpC_24h_sc_PB1_boundary_df$start))
HBEpC_24h_sc_PB1_boundary_df$stop <- as.numeric(as.character(HBEpC_24h_sc_PB1_boundary_df$stop))
save(HBEpC_24h_sc_PB1_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_HBEpC_24h_PB1_boundaries.RData")
dim(unique(HBEpC_24h_sc_PB1_boundary_df[,1:2])) 

# PA
HBEpC_24h_PA_boundary3 <- HBEpC_24h_PA_boundary3[rownames(HBEpC_24h_PA_boundary5), ]
HBEpC_24h_sc_PA_boundary <- list()
for(i in 1:nrow(HBEpC_24h_PA_boundary5)) {
  boundary5 <- data.frame(t(HBEpC_24h_PA_boundary5[i, HBEpC_24h_PA_boundary5[i, ] != "na"]))
  boundary3 <- data.frame(t(HBEpC_24h_PA_boundary3[i, HBEpC_24h_PA_boundary3[i, ] != "na"]))
  HBEpC_24h_sc_PA_boundary[[i]] <- data.frame("start"=boundary5, 
                                              "stop"=boundary3,
                                              "cell"=rownames(HBEpC_24h_PA_boundary5)[i])
  colnames(HBEpC_24h_sc_PA_boundary[[i]])[1:2] <- c("start", "stop")
}
# Because of a bug in rbind, we have to make the first dataframe have more than one row if any of the dataframe in the rest has more than one row
# Do this manually... Move a dataframe with more than one row to the first
HBEpC_24h_sc_PA_boundary_1 <- HBEpC_24h_sc_PA_boundary[[1]]
HBEpC_24h_sc_PA_boundary[[1]] <- HBEpC_24h_sc_PA_boundary[[2]]
HBEpC_24h_sc_PA_boundary[[2]] <- HBEpC_24h_sc_PA_boundary_1
HBEpC_24h_sc_PA_boundary_df <- do.call("rbind", HBEpC_24h_sc_PA_boundary)
HBEpC_24h_sc_PA_boundary_df$start <- as.numeric(as.character(HBEpC_24h_sc_PA_boundary_df$start))
HBEpC_24h_sc_PA_boundary_df$stop <- as.numeric(as.character(HBEpC_24h_sc_PA_boundary_df$stop))
save(HBEpC_24h_sc_PA_boundary_df, 
     file="DIP_boundary_10X-bulk-stock/before-filter/sc_HBEpC_24h_PA_boundaries.RData")
dim(unique(HBEpC_24h_sc_PA_boundary_df[,1:2]))

```



#### Extract the boundary combinations for cellls passed QC ####
```{r warning=FALSE, message=FALSE}
s2_cells <- colnames(s2)
s3_cells <- colnames(s3)
s4_cells <- colnames(s4)
s6_cells <- colnames(s6)
s7_cells <- colnames(s7)
s8_cells <- colnames(s8)


## A549 6hpi
A549_6h_sc_PB2_boundary_df$cell <- as.character(A549_6h_sc_PB2_boundary_df$cell)
A549_6h_sc_PB2_boundary_df_QCC <- A549_6h_sc_PB2_boundary_df[A549_6h_sc_PB2_boundary_df$cell %in% s2_cells, ]
length(unique(A549_6h_sc_PB2_boundary_df_QCC$cell)) 

A549_6h_sc_PB1_boundary_df$cell <- as.character(A549_6h_sc_PB1_boundary_df$cell)
A549_6h_sc_PB1_boundary_df_QCC <- A549_6h_sc_PB1_boundary_df[A549_6h_sc_PB1_boundary_df$cell %in% s2_cells, ]
length(unique(A549_6h_sc_PB1_boundary_df_QCC$cell)) 

A549_6h_sc_PA_boundary_df$cell <- as.character(A549_6h_sc_PA_boundary_df$cell)
A549_6h_sc_PA_boundary_df_QCC <- A549_6h_sc_PA_boundary_df[A549_6h_sc_PA_boundary_df$cell %in% s2_cells, ]
length(unique(A549_6h_sc_PA_boundary_df_QCC$cell)) 


## A549 12hpi
A549_12h_sc_PB2_boundary_df$cell <- as.character(A549_12h_sc_PB2_boundary_df$cell)
A549_12h_sc_PB2_boundary_df_QCC <- A549_12h_sc_PB2_boundary_df[A549_12h_sc_PB2_boundary_df$cell %in% s3_cells, ]
length(unique(A549_12h_sc_PB2_boundary_df_QCC$cell)) 

A549_12h_sc_PB1_boundary_df$cell <- as.character(A549_12h_sc_PB1_boundary_df$cell)
A549_12h_sc_PB1_boundary_df_QCC <- A549_12h_sc_PB1_boundary_df[A549_12h_sc_PB1_boundary_df$cell %in% s3_cells, ]
length(unique(A549_12h_sc_PB1_boundary_df_QCC$cell)) 

A549_12h_sc_PA_boundary_df$cell <- as.character(A549_12h_sc_PA_boundary_df$cell)
A549_12h_sc_PA_boundary_df_QCC <- A549_12h_sc_PA_boundary_df[A549_12h_sc_PA_boundary_df$cell %in% s3_cells, ]
length(unique(A549_12h_sc_PA_boundary_df_QCC$cell)) 


## A549 24hpi
A549_24h_sc_PB2_boundary_df$cell <- as.character(A549_24h_sc_PB2_boundary_df$cell)
A549_24h_sc_PB2_boundary_df_QCC <- A549_24h_sc_PB2_boundary_df[A549_24h_sc_PB2_boundary_df$cell %in% s4_cells, ]
length(unique(A549_24h_sc_PB2_boundary_df_QCC$cell)) 

A549_24h_sc_PB1_boundary_df$cell <- as.character(A549_24h_sc_PB1_boundary_df$cell)
A549_24h_sc_PB1_boundary_df_QCC <- A549_24h_sc_PB1_boundary_df[A549_24h_sc_PB1_boundary_df$cell %in% s4_cells, ]
length(unique(A549_24h_sc_PB1_boundary_df_QCC$cell)) 

A549_24h_sc_PA_boundary_df$cell <- as.character(A549_24h_sc_PA_boundary_df$cell)
A549_24h_sc_PA_boundary_df_QCC <- A549_24h_sc_PA_boundary_df[A549_24h_sc_PA_boundary_df$cell %in% s4_cells, ]
length(unique(A549_24h_sc_PA_boundary_df_QCC$cell)) 


## HBEpC 6hpi
HBEpC_6h_sc_PB2_boundary_df$cell <- as.character(HBEpC_6h_sc_PB2_boundary_df$cell)
HBEpC_6h_sc_PB2_boundary_df_QCC <- HBEpC_6h_sc_PB2_boundary_df[HBEpC_6h_sc_PB2_boundary_df$cell %in% s6_cells, ]
length(unique(HBEpC_6h_sc_PB2_boundary_df_QCC$cell)) 

HBEpC_6h_sc_PB1_boundary_df$cell <- as.character(HBEpC_6h_sc_PB1_boundary_df$cell)
HBEpC_6h_sc_PB1_boundary_df_QCC <- HBEpC_6h_sc_PB1_boundary_df[HBEpC_6h_sc_PB1_boundary_df$cell %in% s6_cells, ]
length(unique(HBEpC_6h_sc_PB1_boundary_df_QCC$cell)) 

HBEpC_6h_sc_PA_boundary_df$cell <- as.character(HBEpC_6h_sc_PA_boundary_df$cell)
HBEpC_6h_sc_PA_boundary_df_QCC <- HBEpC_6h_sc_PA_boundary_df[HBEpC_6h_sc_PA_boundary_df$cell %in% s6_cells, ]
length(unique(HBEpC_6h_sc_PA_boundary_df_QCC$cell)) 


## HBEpC 12hpi
HBEpC_12h_sc_PB2_boundary_df$cell <- as.character(HBEpC_12h_sc_PB2_boundary_df$cell)
HBEpC_12h_sc_PB2_boundary_df_QCC <- HBEpC_12h_sc_PB2_boundary_df[HBEpC_12h_sc_PB2_boundary_df$cell %in% s7_cells, ]
length(unique(HBEpC_12h_sc_PB2_boundary_df_QCC$cell)) 

HBEpC_12h_sc_PB1_boundary_df$cell <- as.character(HBEpC_12h_sc_PB1_boundary_df$cell)
HBEpC_12h_sc_PB1_boundary_df_QCC <- HBEpC_12h_sc_PB1_boundary_df[HBEpC_12h_sc_PB1_boundary_df$cell %in% s7_cells, ]
length(unique(HBEpC_12h_sc_PB1_boundary_df_QCC$cell)) 

HBEpC_12h_sc_PA_boundary_df$cell <- as.character(HBEpC_12h_sc_PA_boundary_df$cell)
HBEpC_12h_sc_PA_boundary_df_QCC <- HBEpC_12h_sc_PA_boundary_df[HBEpC_12h_sc_PA_boundary_df$cell %in% s7_cells, ]
length(unique(HBEpC_12h_sc_PA_boundary_df_QCC$cell)) 


## HBEpC 24hpi
HBEpC_24h_sc_PB2_boundary_df$cell <- as.character(HBEpC_24h_sc_PB2_boundary_df$cell)
HBEpC_24h_sc_PB2_boundary_df_QCC <- HBEpC_24h_sc_PB2_boundary_df[HBEpC_24h_sc_PB2_boundary_df$cell %in% s8_cells, ]
length(unique(HBEpC_24h_sc_PB2_boundary_df_QCC$cell)) 

HBEpC_24h_sc_PB1_boundary_df$cell <- as.character(HBEpC_24h_sc_PB1_boundary_df$cell)
HBEpC_24h_sc_PB1_boundary_df_QCC <- HBEpC_24h_sc_PB1_boundary_df[HBEpC_24h_sc_PB1_boundary_df$cell %in% s8_cells, ]
length(unique(HBEpC_24h_sc_PB1_boundary_df_QCC$cell)) 

HBEpC_24h_sc_PA_boundary_df$cell <- as.character(HBEpC_24h_sc_PA_boundary_df$cell)
HBEpC_24h_sc_PA_boundary_df_QCC <- HBEpC_24h_sc_PA_boundary_df[HBEpC_24h_sc_PA_boundary_df$cell %in% s8_cells, ]
length(unique(HBEpC_24h_sc_PA_boundary_df_QCC$cell)) 


A549_6h_sc_PB2_boundary_df <- A549_6h_sc_PB2_boundary_df_QCC
A549_6h_sc_PB1_boundary_df <- A549_6h_sc_PB1_boundary_df_QCC
A549_6h_sc_PA_boundary_df <- A549_6h_sc_PA_boundary_df_QCC

A549_12h_sc_PB2_boundary_df <- A549_12h_sc_PB2_boundary_df_QCC
A549_12h_sc_PB1_boundary_df <- A549_12h_sc_PB1_boundary_df_QCC
A549_12h_sc_PA_boundary_df <- A549_12h_sc_PA_boundary_df_QCC

A549_24h_sc_PB2_boundary_df <- A549_24h_sc_PB2_boundary_df_QCC
A549_24h_sc_PB1_boundary_df <- A549_24h_sc_PB1_boundary_df_QCC
A549_24h_sc_PA_boundary_df <- A549_24h_sc_PA_boundary_df_QCC

HBEpC_6h_sc_PB2_boundary_df <- HBEpC_6h_sc_PB2_boundary_df_QCC
HBEpC_6h_sc_PB1_boundary_df <- HBEpC_6h_sc_PB1_boundary_df_QCC
HBEpC_6h_sc_PA_boundary_df <- HBEpC_6h_sc_PA_boundary_df_QCC

HBEpC_12h_sc_PB2_boundary_df <- HBEpC_12h_sc_PB2_boundary_df_QCC
HBEpC_12h_sc_PB1_boundary_df <- HBEpC_12h_sc_PB1_boundary_df_QCC
HBEpC_12h_sc_PA_boundary_df <- HBEpC_12h_sc_PA_boundary_df_QCC

HBEpC_24h_sc_PB2_boundary_df <- HBEpC_24h_sc_PB2_boundary_df_QCC
HBEpC_24h_sc_PB1_boundary_df <- HBEpC_24h_sc_PB1_boundary_df_QCC
HBEpC_24h_sc_PA_boundary_df <- HBEpC_24h_sc_PA_boundary_df_QCC


# double check the accuracy of the extraction
sum(!unique(A549_6h_sc_PB2_boundary_df$cell) %in% s2_cells)
sum(!unique(A549_6h_sc_PB1_boundary_df$cell) %in% s2_cells)
sum(!unique(A549_6h_sc_PA_boundary_df$cell) %in% s2_cells) 

sum(!unique(A549_12h_sc_PB2_boundary_df$cell) %in% s3_cells)
sum(!unique(A549_12h_sc_PB1_boundary_df$cell) %in% s3_cells)
sum(!unique(A549_12h_sc_PA_boundary_df$cell) %in% s3_cells) 

sum(!unique(A549_24h_sc_PB2_boundary_df$cell) %in% s4_cells)
sum(!unique(A549_24h_sc_PB1_boundary_df$cell) %in% s4_cells)
sum(!unique(A549_24h_sc_PA_boundary_df$cell) %in% s4_cells) 

sum(!unique(HBEpC_6h_sc_PB2_boundary_df$cell) %in% s6_cells)
sum(!unique(HBEpC_6h_sc_PB1_boundary_df$cell) %in% s6_cells)
sum(!unique(HBEpC_6h_sc_PA_boundary_df$cell) %in% s6_cells) 

sum(!unique(HBEpC_12h_sc_PB2_boundary_df$cell) %in% s7_cells)
sum(!unique(HBEpC_12h_sc_PB1_boundary_df$cell) %in% s7_cells)
sum(!unique(HBEpC_12h_sc_PA_boundary_df$cell) %in% s7_cells) 

sum(!unique(HBEpC_24h_sc_PB2_boundary_df$cell) %in% s8_cells) 
sum(!unique(HBEpC_24h_sc_PB1_boundary_df$cell) %in% s8_cells) 
sum(!unique(HBEpC_24h_sc_PA_boundary_df$cell) %in% s8_cells) 

```



#### Plot the frequency of boundary combinations in single cell data ####
```{r warning=FALSE, message=FALSE}
## A549 6h ##
# PB2
A549_6h_sc_PB2_boundary_combined <- 
  data.frame("boundaries"=paste(A549_6h_sc_PB2_boundary_df$start, 
                                A549_6h_sc_PB2_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(A549_6h_sc_PB2_boundary_df)))
A549_6h_sc_PB2_boundary_combined <- 
  A549_6h_sc_PB2_boundary_combined[order(A549_6h_sc_PB2_boundary_combined$boundaries), ]

ggplot(A549_6h_sc_PB2_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

# PB1
A549_6h_sc_PB1_boundary_combined <- 
  data.frame("boundaries"=paste(A549_6h_sc_PB1_boundary_df$start, 
                                A549_6h_sc_PB1_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(A549_6h_sc_PB1_boundary_df)))
A549_6h_sc_PB1_boundary_combined <- 
  A549_6h_sc_PB1_boundary_combined[order(A549_6h_sc_PB1_boundary_combined$boundaries), ]

ggplot(A549_6h_sc_PB1_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

# PA
A549_6h_sc_PA_boundary_combined <- 
  data.frame("boundaries"=paste(A549_6h_sc_PA_boundary_df$start, 
                                A549_6h_sc_PA_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(A549_6h_sc_PA_boundary_df)))
A549_6h_sc_PA_boundary_combined <- 
  A549_6h_sc_PA_boundary_combined[order(A549_6h_sc_PA_boundary_combined$boundaries), ]

ggplot(A549_6h_sc_PA_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))


## A549 12h ##
# PB2
A549_12h_sc_PB2_boundary_combined <- 
  data.frame("boundaries"=paste(A549_12h_sc_PB2_boundary_df$start, 
                                A549_12h_sc_PB2_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(A549_12h_sc_PB2_boundary_df)))
A549_12h_sc_PB2_boundary_combined <- 
  A549_12h_sc_PB2_boundary_combined[order(A549_12h_sc_PB2_boundary_combined$boundaries), ]

ggplot(A549_12h_sc_PB2_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

# PB1
A549_12h_sc_PB1_boundary_combined <- 
  data.frame("boundaries"=paste(A549_12h_sc_PB1_boundary_df$start, 
                                A549_12h_sc_PB1_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(A549_12h_sc_PB1_boundary_df)))
A549_12h_sc_PB1_boundary_combined <- 
  A549_12h_sc_PB1_boundary_combined[order(A549_12h_sc_PB1_boundary_combined$boundaries), ]

ggplot(A549_12h_sc_PB1_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

# PA
A549_12h_sc_PA_boundary_combined <- 
  data.frame("boundaries"=paste(A549_12h_sc_PA_boundary_df$start, 
                                A549_12h_sc_PA_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(A549_12h_sc_PA_boundary_df)))
A549_12h_sc_PA_boundary_combined <- 
  A549_12h_sc_PA_boundary_combined[order(A549_12h_sc_PA_boundary_combined$boundaries), ]

ggplot(A549_12h_sc_PA_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))


## A549 24h ##
# PB2
A549_24h_sc_PB2_boundary_combined <- 
  data.frame("boundaries"=paste(A549_24h_sc_PB2_boundary_df$start, 
                                A549_24h_sc_PB2_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(A549_24h_sc_PB2_boundary_df)))
A549_24h_sc_PB2_boundary_combined <- 
  A549_24h_sc_PB2_boundary_combined[order(A549_24h_sc_PB2_boundary_combined$boundaries), ]

ggplot(A549_24h_sc_PB2_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=14))

# PB1
A549_24h_sc_PB1_boundary_combined <- 
  data.frame("boundaries"=paste(A549_24h_sc_PB1_boundary_df$start, 
                                A549_24h_sc_PB1_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(A549_24h_sc_PB1_boundary_df)))
A549_24h_sc_PB1_boundary_combined <- 
  A549_24h_sc_PB1_boundary_combined[order(A549_24h_sc_PB1_boundary_combined$boundaries), ]

ggplot(A549_24h_sc_PB1_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=14))

# PA
A549_24h_sc_PA_boundary_combined <- 
  data.frame("boundaries"=paste(A549_24h_sc_PA_boundary_df$start, 
                                A549_24h_sc_PA_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(A549_24h_sc_PA_boundary_df)))
A549_24h_sc_PA_boundary_combined <- 
  A549_24h_sc_PA_boundary_combined[order(A549_24h_sc_PA_boundary_combined$boundaries), ]

ggplot(A549_24h_sc_PA_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))


## HBEpC 6h ##
# PB2
HBEpC_6h_sc_PB2_boundary_combined <- 
  data.frame("boundaries"=paste(HBEpC_6h_sc_PB2_boundary_df$start, 
                                HBEpC_6h_sc_PB2_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(HBEpC_6h_sc_PB2_boundary_df)))
HBEpC_6h_sc_PB2_boundary_combined <- 
  HBEpC_6h_sc_PB2_boundary_combined[order(HBEpC_6h_sc_PB2_boundary_combined$boundaries), ]

ggplot(HBEpC_6h_sc_PB2_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

# PB1
HBEpC_6h_sc_PB1_boundary_combined <- 
  data.frame("boundaries"=paste(HBEpC_6h_sc_PB1_boundary_df$start, 
                                HBEpC_6h_sc_PB1_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(HBEpC_6h_sc_PB1_boundary_df)))
HBEpC_6h_sc_PB1_boundary_combined <- 
  HBEpC_6h_sc_PB1_boundary_combined[order(HBEpC_6h_sc_PB1_boundary_combined$boundaries), ]

ggplot(HBEpC_6h_sc_PB1_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

# PA
HBEpC_6h_sc_PA_boundary_combined <- 
  data.frame("boundaries"=paste(HBEpC_6h_sc_PA_boundary_df$start, 
                                HBEpC_6h_sc_PA_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(HBEpC_6h_sc_PA_boundary_df)))
HBEpC_6h_sc_PA_boundary_combined <- 
  HBEpC_6h_sc_PA_boundary_combined[order(HBEpC_6h_sc_PA_boundary_combined$boundaries), ]

ggplot(HBEpC_6h_sc_PA_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=14))


## HBEpC 12h ##
# PB2
HBEpC_12h_sc_PB2_boundary_combined <- 
  data.frame("boundaries"=paste(HBEpC_12h_sc_PB2_boundary_df$start, 
                                HBEpC_12h_sc_PB2_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(HBEpC_12h_sc_PB2_boundary_df)))
HBEpC_12h_sc_PB2_boundary_combined <- 
  HBEpC_12h_sc_PB2_boundary_combined[order(HBEpC_12h_sc_PB2_boundary_combined$boundaries), ]

ggplot(HBEpC_12h_sc_PB2_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

# PB1
HBEpC_12h_sc_PB1_boundary_combined <- 
  data.frame("boundaries"=paste(HBEpC_12h_sc_PB1_boundary_df$start, 
                                HBEpC_12h_sc_PB1_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(HBEpC_12h_sc_PB1_boundary_df)))
HBEpC_12h_sc_PB1_boundary_combined <- 
  HBEpC_12h_sc_PB1_boundary_combined[order(HBEpC_12h_sc_PB1_boundary_combined$boundaries), ]

ggplot(HBEpC_12h_sc_PB1_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

# PA
HBEpC_12h_sc_PA_boundary_combined <- 
  data.frame("boundaries"=paste(HBEpC_12h_sc_PA_boundary_df$start, 
                                HBEpC_12h_sc_PA_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(HBEpC_12h_sc_PA_boundary_df)))
HBEpC_12h_sc_PA_boundary_combined <- 
  HBEpC_12h_sc_PA_boundary_combined[order(HBEpC_12h_sc_PA_boundary_combined$boundaries), ]

ggplot(HBEpC_12h_sc_PA_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))


## HBEpC 24h ##
# PB2
HBEpC_24h_sc_PB2_boundary_combined <- 
  data.frame("boundaries"=paste(HBEpC_24h_sc_PB2_boundary_df$start, 
                                HBEpC_24h_sc_PB2_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(HBEpC_24h_sc_PB2_boundary_df)))
HBEpC_24h_sc_PB2_boundary_combined <- 
  HBEpC_24h_sc_PB2_boundary_combined[order(HBEpC_24h_sc_PB2_boundary_combined$boundaries), ]

ggplot(HBEpC_24h_sc_PB2_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

# PB1
HBEpC_24h_sc_PB1_boundary_combined <- 
  data.frame("boundaries"=paste(HBEpC_24h_sc_PB1_boundary_df$start, 
                                HBEpC_24h_sc_PB1_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(HBEpC_24h_sc_PB1_boundary_df)))
HBEpC_24h_sc_PB1_boundary_combined <- 
  HBEpC_24h_sc_PB1_boundary_combined[order(HBEpC_24h_sc_PB1_boundary_combined$boundaries), ]

ggplot(HBEpC_24h_sc_PB1_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

# PA
HBEpC_24h_sc_PA_boundary_combined <- 
  data.frame("boundaries"=paste(HBEpC_24h_sc_PA_boundary_df$start, 
                                HBEpC_24h_sc_PA_boundary_df$stop,
                                sep="-"),
             "count"=rep(1, nrow(HBEpC_24h_sc_PA_boundary_df)))
HBEpC_24h_sc_PA_boundary_combined <- 
  HBEpC_24h_sc_PA_boundary_combined[order(HBEpC_24h_sc_PA_boundary_combined$boundaries), ]

ggplot(HBEpC_24h_sc_PA_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

```



#### Plot the frequency of boundary combinations in PR8 stock ####
```{r warning=FALSE, message=FALSE}
# PB2
stock_PB2_boundary_combined <- 
  data.frame("boundaries"=paste(stock_PB2_boundary$start, 
                                stock_PB2_boundary$end,
                                sep="-"),
             "count"=rep(1, nrow(stock_PB2_boundary)))
stock_PB2_boundary_combined <- 
  stock_PB2_boundary_combined[stock_PB2_boundary_combined$boundaries != "na-na", ]
stock_PB2_boundary_combined <- 
  stock_PB2_boundary_combined[order(stock_PB2_boundary_combined$boundaries), ]

ggplot(stock_PB2_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=10))

# PB1
stock_PB1_boundary_combined <- 
  data.frame("boundaries"=paste(stock_PB1_boundary$start, 
                                stock_PB1_boundary$end,
                                sep="-"),
             "count"=rep(1, nrow(stock_PB1_boundary)))
stock_PB1_boundary_combined <-
  stock_PB1_boundary_combined[stock_PB1_boundary_combined$boundaries != "na-na", ]
stock_PB1_boundary_combined <- 
  stock_PB1_boundary_combined[order(stock_PB1_boundary_combined$boundaries), ]

ggplot(stock_PB1_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=10))

# PA
stock_PA_boundary_combined <- 
  data.frame("boundaries"=paste(stock_PA_boundary$start, 
                                stock_PA_boundary$end,
                                sep="-"),
             "count"=rep(1, nrow(stock_PA_boundary)))
stock_PA_boundary_combined <-
  stock_PA_boundary_combined[stock_PA_boundary_combined$boundaries != "na-na", ]
stock_PA_boundary_combined <- 
  stock_PA_boundary_combined[order(stock_PA_boundary_combined$boundaries), ]

ggplot(stock_PA_boundary_combined, aes(boundaries)) +
  geom_histogram(stat="count", fill="deepskyblue3") +
  scale_y_continuous(trans="log10", 
                     labels=trans_format("log10", math_format(10^.x)),
                     breaks=trans_breaks("log10", function(x) 10^x)) +
  annotation_logticks(base=10, sides="l") +
  theme(panel.background=element_rect(fill="white", colour="grey50"),
        panel.grid.major=element_line(linetype="solid", color="grey90"),
        axis.text=element_text(angle=45, hjust=1),
        text=element_text(size=15))

```



#### Obtain the frequency table for each sample ####
```{r warning=FALSE, message=FALSE, include=TRUE, eval=FALSE}
## Sinlge cell data ##
# A549 6hpi
A549_6h_sc_PB2_boundary_df$start_stop <- paste(A549_6h_sc_PB2_boundary_df$start,
                                               A549_6h_sc_PB2_boundary_df$stop,
                                               sep="-")
A549_6h_sc_PB1_boundary_df$start_stop <- paste(A549_6h_sc_PB1_boundary_df$start,
                                               A549_6h_sc_PB1_boundary_df$stop,
                                               sep="-")
A549_6h_sc_PA_boundary_df$start_stop <- paste(A549_6h_sc_PA_boundary_df$start,
                                              A549_6h_sc_PA_boundary_df$stop,
                                              sep="-")

A549_6h_sc_PB2_boundary_fq <- as.data.frame(table(A549_6h_sc_PB2_boundary_df$start_stop))
A549_6h_sc_PB1_boundary_fq <- as.data.frame(table(A549_6h_sc_PB1_boundary_df$start_stop))
A549_6h_sc_PA_boundary_fq <- as.data.frame(table(A549_6h_sc_PA_boundary_df$start_stop))

colnames(A549_6h_sc_PB2_boundary_fq) <- c("boundary_comb", "freq")
colnames(A549_6h_sc_PB1_boundary_fq) <- c("boundary_comb", "freq")
colnames(A549_6h_sc_PA_boundary_fq) <- c("boundary_comb", "freq")

write.csv(A549_6h_sc_PB2_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/A549_6h_sc_PB2_boundary_freq.csv")
write.csv(A549_6h_sc_PB1_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/A549_6h_sc_PB1_boundary_freq.csv")
write.csv(A549_6h_sc_PA_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/A549_6h_sc_PA_boundary_freq.csv")


# A549 12hpi
A549_12h_sc_PB2_boundary_df$start_stop <- paste(A549_12h_sc_PB2_boundary_df$start,
                                                A549_12h_sc_PB2_boundary_df$stop,
                                                sep="-")
A549_12h_sc_PB1_boundary_df$start_stop <- paste(A549_12h_sc_PB1_boundary_df$start,
                                                A549_12h_sc_PB1_boundary_df$stop,
                                                sep="-")
A549_12h_sc_PA_boundary_df$start_stop <- paste(A549_12h_sc_PA_boundary_df$start,
                                               A549_12h_sc_PA_boundary_df$stop,
                                               sep="-")

A549_12h_sc_PB2_boundary_fq <- as.data.frame(table(A549_12h_sc_PB2_boundary_df$start_stop))
A549_12h_sc_PB1_boundary_fq <- as.data.frame(table(A549_12h_sc_PB1_boundary_df$start_stop))
A549_12h_sc_PA_boundary_fq <- as.data.frame(table(A549_12h_sc_PA_boundary_df$start_stop))

colnames(A549_12h_sc_PB2_boundary_fq) <- c("boundary_comb", "freq")
colnames(A549_12h_sc_PB1_boundary_fq) <- c("boundary_comb", "freq")
colnames(A549_12h_sc_PA_boundary_fq) <- c("boundary_comb", "freq")

write.csv(A549_12h_sc_PB2_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/A549_12h_sc_PB2_boundary_freq_new.csv")
write.csv(A549_12h_sc_PB1_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/A549_12h_sc_PB1_boundary_freq_new.csv")
write.csv(A549_12h_sc_PA_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/A549_12h_sc_PA_boundary_freq_new.csv")


# A549 24hpi
A549_24h_sc_PB2_boundary_df$start_stop <- paste(A549_24h_sc_PB2_boundary_df$start,
                                                A549_24h_sc_PB2_boundary_df$stop,
                                                sep="-")
A549_24h_sc_PB1_boundary_df$start_stop <- paste(A549_24h_sc_PB1_boundary_df$start,
                                                A549_24h_sc_PB1_boundary_df$stop,
                                                sep="-")
A549_24h_sc_PA_boundary_df$start_stop <- paste(A549_24h_sc_PA_boundary_df$start,
                                               A549_24h_sc_PA_boundary_df$stop,
                                               sep="-")

A549_24h_sc_PB2_boundary_fq <- as.data.frame(table(A549_24h_sc_PB2_boundary_df$start_stop))
A549_24h_sc_PB1_boundary_fq <- as.data.frame(table(A549_24h_sc_PB1_boundary_df$start_stop))
A549_24h_sc_PA_boundary_fq <- as.data.frame(table(A549_24h_sc_PA_boundary_df$start_stop))

colnames(A549_24h_sc_PB2_boundary_fq) <- c("boundary_comb", "freq")
colnames(A549_24h_sc_PB1_boundary_fq) <- c("boundary_comb", "freq")
colnames(A549_24h_sc_PA_boundary_fq) <- c("boundary_comb", "freq")

write.csv(A549_24h_sc_PB2_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/A549_24h_sc_PB2_boundary_freq.csv")
write.csv(A549_24h_sc_PB1_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/A549_24h_sc_PB1_boundary_freq.csv")
write.csv(A549_24h_sc_PA_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/A549_24h_sc_PA_boundary_freq.csv")


# HBEpC 6hpi
HBEpC_6h_sc_PB2_boundary_df$start_stop <- paste(HBEpC_6h_sc_PB2_boundary_df$start,
                                                HBEpC_6h_sc_PB2_boundary_df$stop,
                                                sep="-")
HBEpC_6h_sc_PB1_boundary_df$start_stop <- paste(HBEpC_6h_sc_PB1_boundary_df$start,
                                                HBEpC_6h_sc_PB1_boundary_df$stop,
                                                sep="-")
HBEpC_6h_sc_PA_boundary_df$start_stop <- paste(HBEpC_6h_sc_PA_boundary_df$start,
                                               HBEpC_6h_sc_PA_boundary_df$stop,
                                               sep="-")

HBEpC_6h_sc_PB2_boundary_fq <- as.data.frame(table(HBEpC_6h_sc_PB2_boundary_df$start_stop))
HBEpC_6h_sc_PB1_boundary_fq <- as.data.frame(table(HBEpC_6h_sc_PB1_boundary_df$start_stop))
HBEpC_6h_sc_PA_boundary_fq <- as.data.frame(table(HBEpC_6h_sc_PA_boundary_df$start_stop))

colnames(HBEpC_6h_sc_PB2_boundary_fq) <- c("boundary_comb", "freq")
colnames(HBEpC_6h_sc_PB1_boundary_fq) <- c("boundary_comb", "freq")
colnames(HBEpC_6h_sc_PA_boundary_fq) <- c("boundary_comb", "freq")

write.csv(HBEpC_6h_sc_PB2_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/HBEpC_6h_sc_PB2_boundary_freq.csv")
write.csv(HBEpC_6h_sc_PB1_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/HBEpC_6h_sc_PB1_boundary_freq.csv")
write.csv(HBEpC_6h_sc_PA_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/HBEpC_6h_sc_PA_boundary_freq.csv")


# HBEpC 12hpi
HBEpC_12h_sc_PB2_boundary_df$start_stop <- paste(HBEpC_12h_sc_PB2_boundary_df$start,
                                                 HBEpC_12h_sc_PB2_boundary_df$stop,
                                                 sep="-")
HBEpC_12h_sc_PB1_boundary_df$start_stop <- paste(HBEpC_12h_sc_PB1_boundary_df$start,
                                                 HBEpC_12h_sc_PB1_boundary_df$stop,
                                                 sep="-")
HBEpC_12h_sc_PA_boundary_df$start_stop <- paste(HBEpC_12h_sc_PA_boundary_df$start,
                                                HBEpC_12h_sc_PA_boundary_df$stop,
                                                sep="-")

HBEpC_12h_sc_PB2_boundary_fq <- as.data.frame(table(HBEpC_12h_sc_PB2_boundary_df$start_stop))
HBEpC_12h_sc_PB1_boundary_fq <- as.data.frame(table(HBEpC_12h_sc_PB1_boundary_df$start_stop))
HBEpC_12h_sc_PA_boundary_fq <- as.data.frame(table(HBEpC_12h_sc_PA_boundary_df$start_stop))

colnames(HBEpC_12h_sc_PB2_boundary_fq) <- c("boundary_comb", "freq")
colnames(HBEpC_12h_sc_PB1_boundary_fq) <- c("boundary_comb", "freq")
colnames(HBEpC_12h_sc_PA_boundary_fq) <- c("boundary_comb", "freq")

write.csv(HBEpC_12h_sc_PB2_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/HBEpC_12h_sc_PB2_boundary_freq.csv")
write.csv(HBEpC_12h_sc_PB1_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/HBEpC_12h_sc_PB1_boundary_freq.csv")
write.csv(HBEpC_12h_sc_PA_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/HBEpC_12h_sc_PA_boundary_freq.csv")


# HBEpC 24hpi
HBEpC_24h_sc_PB2_boundary_df$start_stop <- paste(HBEpC_24h_sc_PB2_boundary_df$start,
                                                 HBEpC_24h_sc_PB2_boundary_df$stop,
                                                 sep="-")
HBEpC_24h_sc_PB1_boundary_df$start_stop <- paste(HBEpC_24h_sc_PB1_boundary_df$start,
                                                 HBEpC_24h_sc_PB1_boundary_df$stop,
                                                 sep="-")
HBEpC_24h_sc_PA_boundary_df$start_stop <- paste(HBEpC_24h_sc_PA_boundary_df$start,
                                                HBEpC_24h_sc_PA_boundary_df$stop,
                                                sep="-")

HBEpC_24h_sc_PB2_boundary_fq <- as.data.frame(table(HBEpC_24h_sc_PB2_boundary_df$start_stop))
HBEpC_24h_sc_PB1_boundary_fq <- as.data.frame(table(HBEpC_24h_sc_PB1_boundary_df$start_stop))
HBEpC_24h_sc_PA_boundary_fq <- as.data.frame(table(HBEpC_24h_sc_PA_boundary_df$start_stop))

colnames(HBEpC_24h_sc_PB2_boundary_fq) <- c("boundary_comb", "freq")
colnames(HBEpC_24h_sc_PB1_boundary_fq) <- c("boundary_comb", "freq")
colnames(HBEpC_24h_sc_PA_boundary_fq) <- c("boundary_comb", "freq")

write.csv(HBEpC_24h_sc_PB2_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/HBEpC_24h_sc_PB2_boundary_freq.csv")
write.csv(HBEpC_24h_sc_PB1_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/HBEpC_24h_sc_PB1_boundary_freq.csv")
write.csv(HBEpC_24h_sc_PA_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/HBEpC_24h_sc_PA_boundary_freq.csv")


## PR8 stock
stock_PB2_boundary$start_stop <- paste(stock_PB2_boundary$start,
                                       stock_PB2_boundary$end,
                                       sep="-")
stock_PB1_boundary$start_stop <- paste(stock_PB1_boundary$start,
                                       stock_PB1_boundary$end,
                                       sep="-")
stock_PA_boundary$start_stop <- paste(stock_PA_boundary$start,
                                      stock_PA_boundary$end,
                                      sep="-")

stock_PB2_boundary_fq <- as.data.frame(table(stock_PB2_boundary$start_stop))
stock_PB1_boundary_fq <- as.data.frame(table(stock_PB1_boundary$start_stop))
stock_PA_boundary_fq <- as.data.frame(table(stock_PA_boundary$start_stop))

colnames(stock_PB2_boundary_fq) <- c("boundary_comb", "freq")
colnames(stock_PB1_boundary_fq) <- c("boundary_comb", "freq")
colnames(stock_PA_boundary_fq) <- c("boundary_comb", "freq")

write.csv(stock_PB2_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/PR8_stock_PB2_boundary_freq.csv")
write.csv(stock_PB1_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/PR8_stock_PB1_boundary_freq.csv")
write.csv(stock_PA_boundary_fq, 
          file="DIP_boundary_10X-bulk-stock/before-filter/Boundaries_frequency_table/PR8_stock_PA_boundary_freq.csv")

```



#### Group boundary coordinates together (within 10nt at both boundaries) and filter out grouped coordinates occurred less than 10 times in each sample ####
# Use python script (DI_boundary_grouping.py)



#### Reload and visualize the grouped boundary coordinates with tile plot ####
```{r warning=FALSE, message=FALSE}
setwd("~/Documents/10XGenomics_BulkRNAseq_qPCR_2017Fall/10X_data_July2018_v2/DIP_boundary_10X-bulk-stock/after-filter/")

## Load data ##
sc_A549_6h <- read.csv("sc_A549_6h_DI_boundary_grouped.csv")
sc_A549_12h <- read.csv("sc_A549_12h_DI_boundary_grouped.csv")
sc_A549_24h <- read.csv("sc_A549_24h_DI_boundary_grouped.csv")
sc_HBEpC_6h <- read.csv("sc_HBEpC_6h_DI_boundary_grouped.csv")
sc_HBEpC_12h <- read.csv("sc_HBEpC_12h_DI_boundary_grouped.csv")
sc_HBEpC_24h <- read.csv("sc_HBEpC_24h_DI_boundary_grouped.csv")
stock <- read.csv("PR8_stock_DI_boundary_grouped.csv")


## Prepare data ##
sc_A549_6h$boundary_idx <- factor(sc_A549_6h$boundary_idx)
sc_A549_12h$boundary_idx <- factor(sc_A549_12h$boundary_idx)
sc_A549_24h$boundary_idx <- factor(sc_A549_24h$boundary_idx)
sc_HBEpC_6h$boundary_idx <- factor(sc_HBEpC_6h$boundary_idx)
sc_HBEpC_12h$boundary_idx <- factor(sc_HBEpC_12h$boundary_idx)
sc_HBEpC_24h$boundary_idx <- factor(sc_HBEpC_24h$boundary_idx)
stock$boundary_idx <- factor(stock$boundary_idx)

# stock
stock_PB2_list <- list()
stock_PB2 <- stock[stock$seg == "PB2", ]
for(l in stock_PB2$boundary_idx) {
  stock_PB2_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (stock_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (stock_PB2$boundary_5p_2[as.numeric(l)] - stock_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (stock_PB2$boundary_3p_1[as.numeric(l)] - stock_PB2$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (stock_PB2$boundary_3p_2[as.numeric(l)] - stock_PB2$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - stock_PB2$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(stock_PB2[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(stock_PB2[as.numeric(l), "freq"], 2341))
}
stock_PB2_full_df <- do.call("rbind", stock_PB2_list)
stock_PB2_full_df$category <- factor(stock_PB2_full_df$category,
                                     levels=c("Retained region", "Boundary start", 
                                              "Deletion", "Boundary end"))

stock_PB1_list <- list()
stock_PB1 <- stock[stock$seg == "PB1", ]
for(l in stock_PB1$boundary_idx) {
  stock_PB1_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (stock_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (stock_PB1$boundary_5p_2[as.numeric(l)] - stock_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (stock_PB1$boundary_3p_1[as.numeric(l)] - stock_PB1$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (stock_PB1$boundary_3p_2[as.numeric(l)] - stock_PB1$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - stock_PB1$boundary_3p_2[as.numeric(l)]  + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(stock_PB1[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(stock_PB1[as.numeric(l), "freq"], 2341))
}
stock_PB1_full_df <- do.call("rbind", stock_PB1_list)
stock_PB1_full_df$category <- factor(stock_PB1_full_df$category,
                                     levels=c("Retained region", "Boundary start", 
                                              "Deletion", "Boundary end"))

stock_PA_list <- list()
stock_PA <- stock[stock$seg == "PA", ]
for(l in stock_PA$boundary_idx) {
  stock_PA_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (stock_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (stock_PA$boundary_5p_2[as.numeric(l)] - stock_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (stock_PA$boundary_3p_1[as.numeric(l)] - stock_PA$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (stock_PA$boundary_3p_2[as.numeric(l)] - stock_PA$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2233 - stock_PA$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2233, 1),
               "boundary_idx"=rep(l, 2233),
               "boundary_comb"=rep(stock_PA[as.numeric(l), "boundary_comb"], 2233),
               "freq"=rep(stock_PA[as.numeric(l), "freq"], 2233))
}
stock_PA_full_df <- do.call("rbind", stock_PA_list)
stock_PA_full_df$category <- factor(stock_PA_full_df$category,
                                    levels=c("Retained region", "Boundary start", 
                                             "Deletion", "Boundary end"))

# A549 6h
sc_A549_6h_PB2_list <- list()
sc_A549_6h_PB2 <- sc_A549_6h[sc_A549_6h$seg == "PB2", ]
for(l in sc_A549_6h_PB2$boundary_idx) {
  sc_A549_6h_PB2_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_A549_6h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_A549_6h_PB2$boundary_5p_2[as.numeric(l)] - sc_A549_6h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_A549_6h_PB2$boundary_3p_1[as.numeric(l)] - sc_A549_6h_PB2$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_A549_6h_PB2$boundary_3p_2[as.numeric(l)] - sc_A549_6h_PB2$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_A549_6h_PB2$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_A549_6h_PB2[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_A549_6h_PB2[as.numeric(l), "freq"], 2341))
}
sc_A549_6h_PB2_full_df <- do.call("rbind", sc_A549_6h_PB2_list)
sc_A549_6h_PB2_full_df$category <- factor(sc_A549_6h_PB2_full_df$category,
                                          levels=c("Retained region", "Boundary start", 
                                                   "Deletion", "Boundary end"))

sc_A549_6h_PB1_list <- list()
sc_A549_6h_PB1 <- sc_A549_6h[sc_A549_6h$seg == "PB1", ]
for(l in sc_A549_6h_PB1$boundary_idx) {
  sc_A549_6h_PB1_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_A549_6h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_A549_6h_PB1$boundary_5p_2[as.numeric(l)] - sc_A549_6h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_A549_6h_PB1$boundary_3p_1[as.numeric(l)] - sc_A549_6h_PB1$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_A549_6h_PB1$boundary_3p_2[as.numeric(l)] - sc_A549_6h_PB1$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_A549_6h_PB1$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_A549_6h_PB1[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_A549_6h_PB1[as.numeric(l), "freq"], 2341))
}
sc_A549_6h_PB1_full_df <- do.call("rbind", sc_A549_6h_PB1_list)
sc_A549_6h_PB1_full_df$category <- factor(sc_A549_6h_PB1_full_df$category,
                                          levels=c("Retained region", "Boundary start", 
                                                   "Deletion", "Boundary end"))

sc_A549_6h_PA_list <- list()
sc_A549_6h_PA <- sc_A549_6h[sc_A549_6h$seg == "PA", ]
for(l in sc_A549_6h_PA$boundary_idx) {
  sc_A549_6h_PA_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_A549_6h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_A549_6h_PA$boundary_5p_2[as.numeric(l)] - sc_A549_6h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_A549_6h_PA$boundary_3p_1[as.numeric(l)] - sc_A549_6h_PA$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_A549_6h_PA$boundary_3p_2[as.numeric(l)] - sc_A549_6h_PA$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2233 - sc_A549_6h_PA$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2233, 1),
               "boundary_idx"=rep(l, 2233),
               "boundary_comb"=rep(sc_A549_6h_PA[as.numeric(l), "boundary_comb"], 2233),
               "freq"=rep(sc_A549_6h_PA[as.numeric(l), "freq"], 2233))
}
sc_A549_6h_PA_full_df <- do.call("rbind", sc_A549_6h_PA_list)
sc_A549_6h_PA_full_df$category <- factor(sc_A549_6h_PA_full_df$category,
                                         levels=c("Retained region", "Boundary start", 
                                                  "Deletion", "Boundary end"))

# A549 12h
sc_A549_12h_PB2_list <- list()
sc_A549_12h_PB2 <- sc_A549_12h[sc_A549_12h$seg == "PB2", ]
for(l in sc_A549_12h_PB2$boundary_idx) {
  sc_A549_12h_PB2_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_A549_12h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_A549_12h_PB2$boundary_5p_2[as.numeric(l)] - sc_A549_12h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_A549_12h_PB2$boundary_3p_1[as.numeric(l)] - sc_A549_12h_PB2$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_A549_12h_PB2$boundary_3p_2[as.numeric(l)] - sc_A549_12h_PB2$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_A549_12h_PB2$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_A549_12h_PB2[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_A549_12h_PB2[as.numeric(l), "freq"], 2341))
}
sc_A549_12h_PB2_full_df <- do.call("rbind", sc_A549_12h_PB2_list)
sc_A549_12h_PB2_full_df$category <- factor(sc_A549_12h_PB2_full_df$category,
                                           levels=c("Retained region", "Boundary start", 
                                                    "Deletion", "Boundary end"))

sc_A549_12h_PB1_list <- list()
sc_A549_12h_PB1 <- sc_A549_12h[sc_A549_12h$seg == "PB1", ]
for(l in sc_A549_12h_PB1$boundary_idx) {
  sc_A549_12h_PB1_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_A549_12h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_A549_12h_PB1$boundary_5p_2[as.numeric(l)] - sc_A549_12h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_A549_12h_PB1$boundary_3p_1[as.numeric(l)] - sc_A549_12h_PB1$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_A549_12h_PB1$boundary_3p_2[as.numeric(l)] - sc_A549_12h_PB1$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_A549_12h_PB1$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_A549_12h_PB1[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_A549_12h_PB1[as.numeric(l), "freq"], 2341))
}
sc_A549_12h_PB1_full_df <- do.call("rbind", sc_A549_12h_PB1_list)
sc_A549_12h_PB1_full_df$category <- factor(sc_A549_12h_PB1_full_df$category,
                                           levels=c("Retained region", "Boundary start", 
                                                    "Deletion", "Boundary end"))

sc_A549_12h_PA_list <- list()
sc_A549_12h_PA <- sc_A549_12h[sc_A549_12h$seg == "PA", ]
for(l in sc_A549_12h_PA$boundary_idx) {
  sc_A549_12h_PA_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_A549_12h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_A549_12h_PA$boundary_5p_2[as.numeric(l)] - sc_A549_12h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_A549_12h_PA$boundary_3p_1[as.numeric(l)] - sc_A549_12h_PA$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_A549_12h_PA$boundary_3p_2[as.numeric(l)] - sc_A549_12h_PA$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2233 - sc_A549_12h_PA$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2233, 1),
               "boundary_idx"=rep(l, 2233),
               "boundary_comb"=rep(sc_A549_12h_PA[as.numeric(l), "boundary_comb"], 2233),
               "freq"=rep(sc_A549_12h_PA[as.numeric(l), "freq"], 2233))
}
sc_A549_12h_PA_full_df <- do.call("rbind", sc_A549_12h_PA_list)
sc_A549_12h_PA_full_df$category <- factor(sc_A549_12h_PA_full_df$category,
                                          levels=c("Retained region", "Boundary start", 
                                                   "Deletion", "Boundary end"))

# A549 24h
sc_A549_24h_PB2_list <- list()
sc_A549_24h_PB2 <- sc_A549_24h[sc_A549_24h$seg == "PB2", ]
for(l in sc_A549_24h_PB2$boundary_idx) {
  sc_A549_24h_PB2_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_A549_24h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_A549_24h_PB2$boundary_5p_2[as.numeric(l)] - sc_A549_24h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_A549_24h_PB2$boundary_3p_1[as.numeric(l)] - sc_A549_24h_PB2$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_A549_24h_PB2$boundary_3p_2[as.numeric(l)] - sc_A549_24h_PB2$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_A549_24h_PB2$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_A549_24h_PB2[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_A549_24h_PB2[as.numeric(l), "freq"], 2341))
}
sc_A549_24h_PB2_full_df <- do.call("rbind", sc_A549_24h_PB2_list)
sc_A549_24h_PB2_full_df$category <- factor(sc_A549_24h_PB2_full_df$category,
                                           levels=c("Retained region", "Boundary start", 
                                                    "Deletion", "Boundary end"))

sc_A549_24h_PB1_list <- list()
sc_A549_24h_PB1 <- sc_A549_24h[sc_A549_24h$seg == "PB1", ]
for(l in sc_A549_24h_PB1$boundary_idx) {
  sc_A549_24h_PB1_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_A549_24h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_A549_24h_PB1$boundary_5p_2[as.numeric(l)] - sc_A549_24h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_A549_24h_PB1$boundary_3p_1[as.numeric(l)] - sc_A549_24h_PB1$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_A549_24h_PB1$boundary_3p_2[as.numeric(l)] - sc_A549_24h_PB1$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_A549_24h_PB1$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_A549_24h_PB1[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_A549_24h_PB1[as.numeric(l), "freq"], 2341))
}
sc_A549_24h_PB1_full_df <- do.call("rbind", sc_A549_24h_PB1_list)
sc_A549_24h_PB1_full_df$category <- factor(sc_A549_24h_PB1_full_df$category,
                                           levels=c("Retained region", "Boundary start", 
                                                    "Deletion", "Boundary end"))

sc_A549_24h_PA_list <- list()
sc_A549_24h_PA <- sc_A549_24h[sc_A549_24h$seg == "PA", ]
for(l in sc_A549_24h_PA$boundary_idx) {
  sc_A549_24h_PA_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_A549_24h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_A549_24h_PA$boundary_5p_2[as.numeric(l)] - sc_A549_24h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_A549_24h_PA$boundary_3p_1[as.numeric(l)] - sc_A549_24h_PA$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_A549_24h_PA$boundary_3p_2[as.numeric(l)] - sc_A549_24h_PA$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2233 - sc_A549_24h_PA$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2233, 1),
               "boundary_idx"=rep(l, 2233),
               "boundary_comb"=rep(sc_A549_24h_PA[as.numeric(l), "boundary_comb"], 2233),
               "freq"=rep(sc_A549_24h_PA[as.numeric(l), "freq"], 2233))
}
sc_A549_24h_PA_full_df <- do.call("rbind", sc_A549_24h_PA_list)
sc_A549_24h_PA_full_df$category <- factor(sc_A549_24h_PA_full_df$category,
                                          levels=c("Retained region", "Boundary start", 
                                                   "Deletion", "Boundary end"))

# HBEpC 6h
sc_HBEpC_6h_PB2_list <- list()
sc_HBEpC_6h_PB2 <- sc_HBEpC_6h[sc_HBEpC_6h$seg == "PB2", ]
for(l in sc_HBEpC_6h_PB2$boundary_idx) {
  sc_HBEpC_6h_PB2_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_HBEpC_6h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_HBEpC_6h_PB2$boundary_5p_2[as.numeric(l)] - sc_HBEpC_6h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_HBEpC_6h_PB2$boundary_3p_1[as.numeric(l)] - sc_HBEpC_6h_PB2$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_HBEpC_6h_PB2$boundary_3p_2[as.numeric(l)] - sc_HBEpC_6h_PB2$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_HBEpC_6h_PB2$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_HBEpC_6h_PB2[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_HBEpC_6h_PB2[as.numeric(l), "freq"], 2341))
}
sc_HBEpC_6h_PB2_full_df <- do.call("rbind", sc_HBEpC_6h_PB2_list)
sc_HBEpC_6h_PB2_full_df$category <- factor(sc_HBEpC_6h_PB2_full_df$category,
                                           levels=c("Retained region", "Boundary start", 
                                                    "Deletion", "Boundary end"))

sc_HBEpC_6h_PB1_list <- list()
sc_HBEpC_6h_PB1 <- sc_HBEpC_6h[sc_HBEpC_6h$seg == "PB1", ]
for(l in sc_HBEpC_6h_PB1$boundary_idx) {
  sc_HBEpC_6h_PB1_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_HBEpC_6h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_HBEpC_6h_PB1$boundary_5p_2[as.numeric(l)] - sc_HBEpC_6h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_HBEpC_6h_PB1$boundary_3p_1[as.numeric(l)] - sc_HBEpC_6h_PB1$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_HBEpC_6h_PB1$boundary_3p_2[as.numeric(l)] - sc_HBEpC_6h_PB1$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_HBEpC_6h_PB1$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_HBEpC_6h_PB1[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_HBEpC_6h_PB1[as.numeric(l), "freq"], 2341))
}
sc_HBEpC_6h_PB1_full_df <- do.call("rbind", sc_HBEpC_6h_PB1_list)
sc_HBEpC_6h_PB1_full_df$category <- factor(sc_HBEpC_6h_PB1_full_df$category,
                                           levels=c("Retained region", "Boundary start", 
                                                    "Deletion", "Boundary end"))

sc_HBEpC_6h_PA_list <- list()
sc_HBEpC_6h_PA <- sc_HBEpC_6h[sc_HBEpC_6h$seg == "PA", ]
for(l in sc_HBEpC_6h_PA$boundary_idx) {
  sc_HBEpC_6h_PA_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_HBEpC_6h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_HBEpC_6h_PA$boundary_5p_2[as.numeric(l)] - sc_HBEpC_6h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_HBEpC_6h_PA$boundary_3p_1[as.numeric(l)] - sc_HBEpC_6h_PA$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_HBEpC_6h_PA$boundary_3p_2[as.numeric(l)] - sc_HBEpC_6h_PA$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2233 - sc_HBEpC_6h_PA$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2233, 1),
               "boundary_idx"=rep(l, 2233),
               "boundary_comb"=rep(sc_HBEpC_6h_PA[as.numeric(l), "boundary_comb"], 2233),
               "freq"=rep(sc_HBEpC_6h_PA[as.numeric(l), "freq"], 2233))
}
sc_HBEpC_6h_PA_full_df <- do.call("rbind", sc_HBEpC_6h_PA_list)
sc_HBEpC_6h_PA_full_df$category <- factor(sc_HBEpC_6h_PA_full_df$category,
                                          levels=c("Retained region", "Boundary start", 
                                                   "Deletion", "Boundary end"))

# HBEpC 12h
sc_HBEpC_12h_PB2_list <- list()
sc_HBEpC_12h_PB2 <- sc_HBEpC_12h[sc_HBEpC_12h$seg == "PB2", ]
for(l in sc_HBEpC_12h_PB2$boundary_idx) {
  sc_HBEpC_12h_PB2_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_HBEpC_12h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_HBEpC_12h_PB2$boundary_5p_2[as.numeric(l)] - sc_HBEpC_12h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_HBEpC_12h_PB2$boundary_3p_1[as.numeric(l)] - sc_HBEpC_12h_PB2$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_HBEpC_12h_PB2$boundary_3p_2[as.numeric(l)] - sc_HBEpC_12h_PB2$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_HBEpC_12h_PB2$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_HBEpC_12h_PB2[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_HBEpC_12h_PB2[as.numeric(l), "freq"], 2341))
}
sc_HBEpC_12h_PB2_full_df <- do.call("rbind", sc_HBEpC_12h_PB2_list)
sc_HBEpC_12h_PB2_full_df$category <- factor(sc_HBEpC_12h_PB2_full_df$category,
                                            levels=c("Retained region", "Boundary start", 
                                                     "Deletion", "Boundary end"))

sc_HBEpC_12h_PB1_list <- list()
sc_HBEpC_12h_PB1 <- sc_HBEpC_12h[sc_HBEpC_12h$seg == "PB1", ]
for(l in sc_HBEpC_12h_PB1$boundary_idx) {
  sc_HBEpC_12h_PB1_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_HBEpC_12h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_HBEpC_12h_PB1$boundary_5p_2[as.numeric(l)] - sc_HBEpC_12h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_HBEpC_12h_PB1$boundary_3p_1[as.numeric(l)] - sc_HBEpC_12h_PB1$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_HBEpC_12h_PB1$boundary_3p_2[as.numeric(l)] - sc_HBEpC_12h_PB1$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_HBEpC_12h_PB1$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_HBEpC_12h_PB1[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_HBEpC_12h_PB1[as.numeric(l), "freq"], 2341))
}
sc_HBEpC_12h_PB1_full_df <- do.call("rbind", sc_HBEpC_12h_PB1_list)
sc_HBEpC_12h_PB1_full_df$category <- factor(sc_HBEpC_12h_PB1_full_df$category,
                                            levels=c("Retained region", "Boundary start", 
                                                     "Deletion", "Boundary end"))

sc_HBEpC_12h_PA_list <- list()
sc_HBEpC_12h_PA <- sc_HBEpC_12h[sc_HBEpC_12h$seg == "PA", ]
for(l in sc_HBEpC_12h_PA$boundary_idx) {
  sc_HBEpC_12h_PA_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_HBEpC_12h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_HBEpC_12h_PA$boundary_5p_2[as.numeric(l)] - sc_HBEpC_12h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_HBEpC_12h_PA$boundary_3p_1[as.numeric(l)] - sc_HBEpC_12h_PA$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_HBEpC_12h_PA$boundary_3p_2[as.numeric(l)] - sc_HBEpC_12h_PA$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2233 - sc_HBEpC_12h_PA$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2233, 1),
               "boundary_idx"=rep(l, 2233),
               "boundary_comb"=rep(sc_HBEpC_12h_PA[as.numeric(l), "boundary_comb"], 2233),
               "freq"=rep(sc_HBEpC_12h_PA[as.numeric(l), "freq"], 2233))
}
sc_HBEpC_12h_PA_full_df <- do.call("rbind", sc_HBEpC_12h_PA_list)
sc_HBEpC_12h_PA_full_df$category <- factor(sc_HBEpC_12h_PA_full_df$category,
                                           levels=c("Retained region", "Boundary start", 
                                                    "Deletion", "Boundary end"))

# HBEpC 24h
sc_HBEpC_24h_PB2_list <- list()
sc_HBEpC_24h_PB2 <- sc_HBEpC_24h[sc_HBEpC_24h$seg == "PB2", ]
for(l in sc_HBEpC_24h_PB2$boundary_idx) {
  sc_HBEpC_24h_PB2_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_HBEpC_24h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_HBEpC_24h_PB2$boundary_5p_2[as.numeric(l)] - sc_HBEpC_24h_PB2$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_HBEpC_24h_PB2$boundary_3p_1[as.numeric(l)] - sc_HBEpC_24h_PB2$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_HBEpC_24h_PB2$boundary_3p_2[as.numeric(l)] - sc_HBEpC_24h_PB2$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_HBEpC_24h_PB2$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_HBEpC_24h_PB2[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_HBEpC_24h_PB2[as.numeric(l), "freq"], 2341))
}
sc_HBEpC_24h_PB2_full_df <- do.call("rbind", sc_HBEpC_24h_PB2_list)
sc_HBEpC_24h_PB2_full_df$category <- factor(sc_HBEpC_24h_PB2_full_df$category,
                                            levels=c("Retained region", "Boundary start", 
                                                     "Deletion", "Boundary end"))

sc_HBEpC_24h_PB1_list <- list()
sc_HBEpC_24h_PB1 <- sc_HBEpC_24h[sc_HBEpC_24h$seg == "PB1", ]
for(l in sc_HBEpC_24h_PB1$boundary_idx) {
  sc_HBEpC_24h_PB1_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_HBEpC_24h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_HBEpC_24h_PB1$boundary_5p_2[as.numeric(l)] - sc_HBEpC_24h_PB1$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_HBEpC_24h_PB1$boundary_3p_1[as.numeric(l)] - sc_HBEpC_24h_PB1$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_HBEpC_24h_PB1$boundary_3p_2[as.numeric(l)] - sc_HBEpC_24h_PB1$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2341 - sc_HBEpC_24h_PB1$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2341, 1),
               "boundary_idx"=rep(l, 2341),
               "boundary_comb"=rep(sc_HBEpC_24h_PB1[as.numeric(l), "boundary_comb"], 2341),
               "freq"=rep(sc_HBEpC_24h_PB1[as.numeric(l), "freq"], 2341))
}
sc_HBEpC_24h_PB1_full_df <- do.call("rbind", sc_HBEpC_24h_PB1_list)
sc_HBEpC_24h_PB1_full_df$category <- factor(sc_HBEpC_24h_PB1_full_df$category,
                                            levels=c("Retained region", "Boundary start", 
                                                     "Deletion", "Boundary end"))

sc_HBEpC_24h_PA_list <- list()
sc_HBEpC_24h_PA <- sc_HBEpC_24h[sc_HBEpC_24h$seg == "PA", ]
for(l in sc_HBEpC_24h_PA$boundary_idx) {
  sc_HBEpC_24h_PA_list[[as.numeric(l)]] <- 
    data.frame("category"=c(rep("Retained region", (sc_HBEpC_24h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Boundary start", (sc_HBEpC_24h_PA$boundary_5p_2[as.numeric(l)] - sc_HBEpC_24h_PA$boundary_5p_1[as.numeric(l)])),
                            rep("Deletion", (sc_HBEpC_24h_PA$boundary_3p_1[as.numeric(l)] - sc_HBEpC_24h_PA$boundary_5p_2[as.numeric(l)]) - 1),
                            rep("Boundary end", (sc_HBEpC_24h_PA$boundary_3p_2[as.numeric(l)] - sc_HBEpC_24h_PA$boundary_3p_1[as.numeric(l)])),
                            rep("Retained region", (2233 - sc_HBEpC_24h_PA$boundary_3p_2[as.numeric(l)] + 1))),
               "position"=seq(1, 2233, 1),
               "boundary_idx"=rep(l, 2233),
               "boundary_comb"=rep(sc_HBEpC_24h_PA[as.numeric(l), "boundary_comb"], 2233),
               "freq"=rep(sc_HBEpC_24h_PA[as.numeric(l), "freq"], 2233))
}
sc_HBEpC_24h_PA_full_df <- do.call("rbind", sc_HBEpC_24h_PA_list)
sc_HBEpC_24h_PA_full_df$category <- factor(sc_HBEpC_24h_PA_full_df$category,
                                           levels=c("Retained region", "Boundary start", 
                                                    "Deletion", "Boundary end"))


## Plot with geom_tile (the defective viral transcripts were ordered by the start position of the 5' boundary) ##
# stock
ggplot(stock_PB2_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

stock_PB1_full_df$boundary_comb <- factor(stock_PB1_full_df$boundary_comb,
                                          levels=as.character(stock_PB1$boundary_comb))
ggplot(stock_PB1_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

stock_PA_full_df$boundary_comb <- factor(stock_PA_full_df$boundary_comb,
                                         levels=as.character(stock_PA$boundary_comb))
ggplot(stock_PA_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

# sc A549 6h
ggplot(sc_A549_6h_PB2_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

ggplot(sc_A549_6h_PB1_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

sc_A549_6h_PA_full_df$boundary_comb <- factor(sc_A549_6h_PA_full_df$boundary_comb,
                                              levels=as.character(sc_A549_6h_PA$boundary_comb))
ggplot(sc_A549_6h_PA_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

# sc A549 12h
ggplot(sc_A549_12h_PB2_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

sc_A549_12h_PB1_full_df$boundary_comb <- factor(sc_A549_12h_PB1_full_df$boundary_comb,
                                                levels=as.character(sc_A549_12h_PB1$boundary_comb))
ggplot(sc_A549_12h_PB1_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

ggplot(sc_A549_12h_PA_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

# sc A549 24h
ggplot(sc_A549_24h_PB2_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

sc_A549_24h_PB1_full_df$boundary_comb <- factor(sc_A549_24h_PB1_full_df$boundary_comb,
                                                levels=as.character(sc_A549_24h_PB1$boundary_comb))
ggplot(sc_A549_24h_PB1_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

sc_A549_24h_PA_full_df$boundary_comb <- factor(sc_A549_24h_PA_full_df$boundary_comb,
                                               levels=as.character(sc_A549_24h_PA$boundary_comb))
ggplot(sc_A549_24h_PA_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

# sc HBEpC 6h
ggplot(sc_HBEpC_6h_PB2_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion \n boundaries", fill="")

ggplot(sc_HBEpC_6h_PB1_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

ggplot(sc_HBEpC_6h_PA_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

# sc HBEpC 12h
ggplot(sc_HBEpC_12h_PB2_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

sc_HBEpC_12h_PB1_full_df$boundary_comb <- factor(sc_HBEpC_12h_PB1_full_df$boundary_comb,
                                                 levels=as.character(sc_HBEpC_12h_PB1$boundary_comb))
ggplot(sc_HBEpC_12h_PB1_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

ggplot(sc_HBEpC_12h_PA_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

# sc HBEpC 24h
ggplot(sc_HBEpC_24h_PB2_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion \n boundaries", fill="")

ggplot(sc_HBEpC_24h_PB1_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

ggplot(sc_HBEpC_24h_PA_full_df, aes(x=position, y=boundary_comb, fill=category)) +
  geom_tile() +
  scale_fill_manual(values=c("slateblue", "gold", "grey85", "gold")) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.position="bottom") +
  scale_x_continuous(breaks=seq(0,2400,200), limits=c(0, 2400)) +
  labs(x="Position", y="Internal deletion boundaries", fill="")

```



#### Obtain the frequency of occurrence of each DI per sample and the relative abundance of each DI and visualize to identify hot spots ####
```{r warning=FALSE, message=FALSE, include=TRUE, eval=FALSE}
# Make sure the library "dplyr" is un-loaded.
### A549 6hpi ###
setwd("~/Documents/10XGenomics_BulkRNAseq_qPCR_2017Fall/10X_data_July2018_v2/DIP_boundary_10X-bulk-stock/after-filter/Plots/")

## Load data
boundary_criteria <- read.csv("../Filtered_Grouped_DI_boundaries_A549_6h.csv")
# Note: the "Filtered_Grouped_DI_boundaries_A549_6h_update3prime.csv" file has 7 columns:
# 1) boundary_comb, 2) grouped_boundary_comb, 3) seg, 4) sample, 5) boundary_idx
# 6) Overlap_stock, 7) Color (needs to be as consistent as possible across samples)

## Filter the single cell data
PB2_boundary_comb <- A549_6h_sc_PB2_boundary_df
PB2_boundary_comb$boundary_comb <- paste(PB2_boundary_comb$start, 
                                         PB2_boundary_comb$stop, sep="^")

PB1_boundary_comb <- A549_6h_sc_PB1_boundary_df
PB1_boundary_comb$boundary_comb <- paste(PB1_boundary_comb$start, 
                                         PB1_boundary_comb$stop, sep="^")

PA_boundary_comb <- A549_6h_sc_PA_boundary_df
PA_boundary_comb$boundary_comb <- paste(PA_boundary_comb$start, 
                                        PA_boundary_comb$stop, sep="^")

PB2_boundary_comb_filtered <- merge(PB2_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB2", ])
PB1_boundary_comb_filtered <- merge(PB1_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB1", ])
PA_boundary_comb_filtered <- merge(PA_boundary_comb, 
                                   boundary_criteria[boundary_criteria$seg == "PA", ])

## Extract the information about the frequency of a boundary combination in the dataset
PB2_filtered_freq <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PB1_filtered_freq <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PA_filtered_freq <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                          "grouped_boundary_comb")
colnames(PB2_filtered_freq)[2] <- "counts"
colnames(PB1_filtered_freq)[2] <- "counts"
colnames(PA_filtered_freq)[2] <- "counts"

## Extraction the information about the number of cells in which a boundary was observed
PB2_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB2_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB2_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB2", -1])
PB2_filtered_bc_info <- merge(PB2_filtered_freq, boundary_criteria_PB2_unique, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info <- merge(PB2_filtered_cfreq, PB2_filtered_bc_info, 
                              by="boundary_idx")
# Note: "counts" represents the number of times a boundary observed in the whole dataset (all the cells)
# "cell_number" represents the number of cells in which a boundary was observed
PB2_filtered_bc_info$Overlap_stock <- factor(PB2_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PB1_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB1_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB1_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB1", -1])
PB1_filtered_bc_info <- merge(PB1_filtered_freq, boundary_criteria_PB1_unique, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info <- merge(PB1_filtered_cfreq, PB1_filtered_bc_info, 
                              by="boundary_idx")
PB1_filtered_bc_info$Overlap_stock <- factor(PB1_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PA_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                "cell_number"=integer())
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PA_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PA_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PA", -1])
PA_filtered_bc_info <- merge(PA_filtered_freq, boundary_criteria_PA_unique, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info <- merge(PA_filtered_cfreq, PA_filtered_bc_info, 
                             by="boundary_idx")
PA_filtered_bc_info$Overlap_stock <- factor(PA_filtered_bc_info$Overlap_stock,
                                            levels=c("Yes", "No"))

## Count the filtered DIs in each cell
PB2_filtered_count <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PB1_filtered_count <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PA_filtered_count <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                           "cell")
# merge above data into a dataframe
Poly_filtered_count <- merge(PB2_filtered_count, PB1_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count <- merge(Poly_filtered_count, PA_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count[is.na(Poly_filtered_count)] <- 0
colnames(Poly_filtered_count) <- c("cell", "PB2_DI_count", "PB1_DI_count", "PA_DI_count")
# export
write.csv(Poly_filtered_count, 
          file="../../../A549_6h/A549_6h_polymerases_DI_filtered-grouped_counts.csv")


## Adjust the x and y axis and re-plot ##
# Adjust the x-aixs to percentage #
s2_virus <- s2[(nrow(s2)-7):nrow(s2), ]
s2_vsum <- as.data.frame(colSums(s2_virus)) 
s2_host <- s2[1:(nrow(s2)-8), ]

# Check if any cell with PB2/PB1/PA undetected
sum(s2_virus[1, ] == 0) 
sum(s2_virus[2, ] == 0) 
sum(s2_virus[3, ] == 0) 
# Thus all the cells have viral reads

# add a colume into bc_info dataframe to represent the percentage of cells with 
# corresponding DI boundaries in cells with viral reads
PB2_filtered_bc_info$cell_percentage <- 100 * PB2_filtered_bc_info$cell_number / ncol(s2)
PB1_filtered_bc_info$cell_percentage <- 100 * PB1_filtered_bc_info$cell_number / ncol(s2)
PA_filtered_bc_info$cell_percentage <- 100 * PA_filtered_bc_info$cell_number / ncol(s2)

# Adjust the y-axis to the median of the DI ratios for corresponding DIs by three methods ##
# 1) by the UMI counts including full-length reads at 3' peak region
peak_counts <- read.csv("../../../A549_6h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                        row.names=1)
rownames(peak_counts) <- c("PB2", "PB1", "PA")
peak_counts <- peak_counts[, colnames(s2)]

# 2) by the total UMI counts for each viral segment
s2_poly <- s2_virus[1:3, ]


# PB2
# "3p": fall into 3' peak range
# "total": total UMI counts for a given segment
PB2_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s2_poly[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB2_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB2_filtered_bc_info <- merge(PB2_filtered_bc_info, PB2_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_3p)
PB2_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_total)
PB2_filtered_bc_info$grouped_boundary_comb <- as.character(PB2_filtered_bc_info$grouped_boundary_comb)
PB2_filtered_bc_info$boundary_idx <- as.factor(PB2_filtered_bc_info$boundary_idx)
PB2_filtered_bc_info$Color <- as.character(PB2_filtered_bc_info$Color)

write.csv(PB2_filtered_bc_info, file="../../../A549_6h/A549_6hpi_PB2_DI_boundary_ratio_master_info.csv")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")


# PB1
PB1_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s2_poly[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB1_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB1_filtered_bc_info <- merge(PB1_filtered_bc_info, PB1_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_3p)
PB1_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_total)
PB1_filtered_bc_info$grouped_boundary_comb <- as.character(PB1_filtered_bc_info$grouped_boundary_comb)
PB1_filtered_bc_info$boundary_idx <- as.factor(PB1_filtered_bc_info$boundary_idx)
PB1_filtered_bc_info$Color <- as.character(PB1_filtered_bc_info$Color)

write.csv(PB1_filtered_bc_info, file="../../../A549_6h/A549_6hpi_PB1_DI_boundary_ratio_master_info.csv")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

# PA
PA_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                       "median_DI_ratio_3p"=numeric(),
                                       "median_DI_ratio_total"=numeric(),
                                       stringsAsFactors=FALSE)
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == 
                                grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s2_poly[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PA_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                     i_DIratio_total_median)
}
PA_filtered_bc_info <- merge(PA_filtered_bc_info, PA_eachDI_eachcell_ratio, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PA_filtered_bc_info$median_DI_ratio_3p)
PA_filtered_bc_info$median_DI_ratio_total <- as.numeric(PA_filtered_bc_info$median_DI_ratio_total)
PA_filtered_bc_info$grouped_boundary_comb <- as.character(PA_filtered_bc_info$grouped_boundary_comb)
PA_filtered_bc_info$boundary_idx <- as.factor(PA_filtered_bc_info$boundary_idx)
PA_filtered_bc_info$Color <- as.character(PA_filtered_bc_info$Color)

write.csv(PA_filtered_bc_info, file="../../../A549_6h/A549_6hpi_PA_DI_boundary_ratio_master_info.csv")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")



### A549 12hpi ###
## Load data
boundary_criteria <- read.csv("../Filtered_Grouped_DI_boundaries_A549_12h.csv")

## Filter the single cell data
PB2_boundary_comb <- A549_12h_sc_PB2_boundary_df
PB2_boundary_comb$boundary_comb <- paste(PB2_boundary_comb$start, 
                                         PB2_boundary_comb$stop, sep="^")

PB1_boundary_comb <- A549_12h_sc_PB1_boundary_df
PB1_boundary_comb$boundary_comb <- paste(PB1_boundary_comb$start, 
                                         PB1_boundary_comb$stop, sep="^")

PA_boundary_comb <- A549_12h_sc_PA_boundary_df
PA_boundary_comb$boundary_comb <- paste(PA_boundary_comb$start, 
                                        PA_boundary_comb$stop, sep="^")

PB2_boundary_comb_filtered <- merge(PB2_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB2", ])
PB1_boundary_comb_filtered <- merge(PB1_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB1", ])
PA_boundary_comb_filtered <- merge(PA_boundary_comb, 
                                   boundary_criteria[boundary_criteria$seg == "PA", ])

## Extract the information about the frequency of a boundary combination in the dataset
PB2_filtered_freq <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PB1_filtered_freq <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PA_filtered_freq <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                          "grouped_boundary_comb")
colnames(PB2_filtered_freq)[2] <- "counts"
colnames(PB1_filtered_freq)[2] <- "counts"
colnames(PA_filtered_freq)[2] <- "counts"

## Extraction the information about the number of cells in which a boundary was observed
PB2_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB2_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB2_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB2", -1])
PB2_filtered_bc_info <- merge(PB2_filtered_freq, boundary_criteria_PB2_unique, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info <- merge(PB2_filtered_cfreq, PB2_filtered_bc_info, 
                              by="boundary_idx")
# Note: "counts" represents the number of times a boundary observed in the whole dataset (all the cells)
# "cell_number" represents the number of cells in which a boundary was observed
PB2_filtered_bc_info$Overlap_stock <- factor(PB2_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PB1_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB1_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB1_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB1", -1])
PB1_filtered_bc_info <- merge(PB1_filtered_freq, boundary_criteria_PB1_unique, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info <- merge(PB1_filtered_cfreq, PB1_filtered_bc_info, 
                              by="boundary_idx")
PB1_filtered_bc_info$Overlap_stock <- factor(PB1_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PA_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                "cell_number"=integer())
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PA_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PA_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PA", -1])
PA_filtered_bc_info <- merge(PA_filtered_freq, boundary_criteria_PA_unique, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info <- merge(PA_filtered_cfreq, PA_filtered_bc_info, 
                             by="boundary_idx")
PA_filtered_bc_info$Overlap_stock <- factor(PA_filtered_bc_info$Overlap_stock,
                                            levels=c("Yes", "No"))

## Count the filtered DIs in each cell
PB2_filtered_count <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PB1_filtered_count <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PA_filtered_count <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                           "cell")
# merge above data into a dataframe
Poly_filtered_count <- merge(PB2_filtered_count, PB1_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count <- merge(Poly_filtered_count, PA_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count[is.na(Poly_filtered_count)] <- 0
colnames(Poly_filtered_count) <- c("cell", "PB2_DI_count", "PB1_DI_count", "PA_DI_count")
# export
write.csv(Poly_filtered_count, 
          file="../../../A549_12h/A549_12h_polymerases_DI_filtered-grouped_counts.csv")


## Adjust the x and y axis and re-plot ##
# Adjust the x-aixs to percentage #
s3_virus <- s3[((nrow(s3)-7) : nrow(s3)), ]
s3_vsum <- as.data.frame(colSums(s3_virus)) 
s3_host <- s3[1:(nrow(s3)-8), ]

# Check if any cell with PB2/PB1/PA undetected
sum(s3_virus[1, ] == 0) 
sum(s3_virus[2, ] == 0) 
sum(s3_virus[3, ] == 0) 
# Thus all the cells have viral reads

# add a colume into bc_info dataframe to represent the percentage of cells with 
# corresponding DI boundaries in cells with viral reads
PB2_filtered_bc_info$cell_percentage <- 100 * PB2_filtered_bc_info$cell_number / ncol(s3)
PB1_filtered_bc_info$cell_percentage <- 100 * PB1_filtered_bc_info$cell_number / ncol(s3)
PA_filtered_bc_info$cell_percentage <- 100 * PA_filtered_bc_info$cell_number / ncol(s3)

# Adjust the y-axis to the median of the DI ratios for corresponding DIs by three methods ##
# 1) by the UMI counts including full and gapped reads at 3' peak region
peak_counts <- read.csv("../../../A549_12h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                        row.names=1)
rownames(peak_counts) <- c("PB2", "PB1", "PA")
peak_counts <- peak_counts[, colnames(s3)]

# 2) by the total UMI counts for each viral segment
s3_poly <- s3_virus[1:3, ]


# PB2
# "3p": fall into 3' peak range
# "total": total UMI counts for a given segment
PB2_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s3_poly[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB2_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB2_filtered_bc_info <- merge(PB2_filtered_bc_info, PB2_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_3p)
PB2_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_total)
PB2_filtered_bc_info$grouped_boundary_comb <- as.character(PB2_filtered_bc_info$grouped_boundary_comb)
PB2_filtered_bc_info$boundary_idx <- as.factor(PB2_filtered_bc_info$boundary_idx)
PB2_filtered_bc_info$Color <- as.character(PB2_filtered_bc_info$Color)

write.csv(PB2_filtered_bc_info, file="../../../A549_12h/A549_12hpi_PB2_DI_boundary_ratio_master_info.csv")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")


# PB1
PB1_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s3_poly[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB1_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB1_filtered_bc_info <- merge(PB1_filtered_bc_info, PB1_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_3p)
PB1_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_total)
PB1_filtered_bc_info$grouped_boundary_comb <- as.character(PB1_filtered_bc_info$grouped_boundary_comb)
PB1_filtered_bc_info$boundary_idx <- as.factor(PB1_filtered_bc_info$boundary_idx)
PB1_filtered_bc_info$Color <- as.character(PB1_filtered_bc_info$Color)

write.csv(PB1_filtered_bc_info, file="../../../A549_12h/A549_12hpi_PB1_DI_boundary_ratio_master_info.csv")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")


# PA
PA_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                       "median_DI_ratio_3p"=numeric(),
                                       "median_DI_ratio_total"=numeric(),
                                       stringsAsFactors=FALSE)
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == 
                                grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s3_poly[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PA_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                     i_DIratio_total_median)
}
PA_filtered_bc_info <- merge(PA_filtered_bc_info, PA_eachDI_eachcell_ratio, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PA_filtered_bc_info$median_DI_ratio_3p)
PA_filtered_bc_info$median_DI_ratio_total <- as.numeric(PA_filtered_bc_info$median_DI_ratio_total)
PA_filtered_bc_info$grouped_boundary_comb <- as.character(PA_filtered_bc_info$grouped_boundary_comb)
PA_filtered_bc_info$boundary_idx <- as.factor(PA_filtered_bc_info$boundary_idx)
PA_filtered_bc_info$Color <- as.character(PA_filtered_bc_info$Color)

write.csv(PA_filtered_bc_info, file="../../../A549_12h/A549_12hpi_PA_DI_boundary_ratio_master_info.csv")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")



### A549 24hpi ###
## Load data
boundary_criteria <- read.csv("../Filtered_Grouped_DI_boundaries_A549_24h.csv")

## Filter the single cell data
PB2_boundary_comb <- A549_24h_sc_PB2_boundary_df
PB2_boundary_comb$boundary_comb <- paste(PB2_boundary_comb$start, 
                                         PB2_boundary_comb$stop, sep="^")

PB1_boundary_comb <- A549_24h_sc_PB1_boundary_df
PB1_boundary_comb$boundary_comb <- paste(PB1_boundary_comb$start, 
                                         PB1_boundary_comb$stop, sep="^")

PA_boundary_comb <- A549_24h_sc_PA_boundary_df
PA_boundary_comb$boundary_comb <- paste(PA_boundary_comb$start, 
                                        PA_boundary_comb$stop, sep="^")

PB2_boundary_comb_filtered <- merge(PB2_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB2", ])
PB1_boundary_comb_filtered <- merge(PB1_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB1", ])
PA_boundary_comb_filtered <- merge(PA_boundary_comb, 
                                   boundary_criteria[boundary_criteria$seg == "PA", ])

## Extract the information about the frequency of a boundary combination in the dataset
PB2_filtered_freq <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PB1_filtered_freq <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PA_filtered_freq <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                          "grouped_boundary_comb")
colnames(PB2_filtered_freq)[2] <- "counts"
colnames(PB1_filtered_freq)[2] <- "counts"
colnames(PA_filtered_freq)[2] <- "counts"

## Extraction the information about the number of cells in which a boundary was observed
PB2_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB2_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB2_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB2", -1])
PB2_filtered_bc_info <- merge(PB2_filtered_freq, boundary_criteria_PB2_unique, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info <- merge(PB2_filtered_cfreq, PB2_filtered_bc_info, 
                              by="boundary_idx")
# Note: "counts" represents the number of times a boundary observed in the whole dataset (all the cells)
# "cell_number" represents the number of cells in which a boundary was observed
PB2_filtered_bc_info$Overlap_stock <- factor(PB2_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PB1_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB1_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB1_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB1", -1])
PB1_filtered_bc_info <- merge(PB1_filtered_freq, boundary_criteria_PB1_unique, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info <- merge(PB1_filtered_cfreq, PB1_filtered_bc_info, 
                              by="boundary_idx")
PB1_filtered_bc_info$Overlap_stock <- factor(PB1_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PA_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                "cell_number"=integer())
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PA_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PA_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PA", -1])
PA_filtered_bc_info <- merge(PA_filtered_freq, boundary_criteria_PA_unique, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info <- merge(PA_filtered_cfreq, PA_filtered_bc_info, 
                             by="boundary_idx")
PA_filtered_bc_info$Overlap_stock <- factor(PA_filtered_bc_info$Overlap_stock,
                                            levels=c("Yes", "No"))

## Count the filtered DIs in each cell
PB2_filtered_count <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PB1_filtered_count <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PA_filtered_count <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                           "cell")
# merge above data into a dataframe
Poly_filtered_count <- merge(PB2_filtered_count, PB1_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count <- merge(Poly_filtered_count, PA_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count[is.na(Poly_filtered_count)] <- 0
colnames(Poly_filtered_count) <- c("cell", "PB2_DI_count", "PB1_DI_count", "PA_DI_count")
# export
write.csv(Poly_filtered_count, 
          file="../../../A549_24h/A549_24h_polymerases_DI_filtered-grouped_counts.csv")


## Adjust the x and y axis and re-plot ##
# Adjust the x-aixs to percentage #
s4_virus <- s4[((nrow(s4)-7):nrow(s4)), ]
s4_vsum <- as.data.frame(colSums(s4_virus)) 
s4_host <- s4[1:(nrow(s4)-8), ]

# Check if any cell with PB2/PB1/PA undetected
sum(s4_virus[1, ] == 0) 
sum(s4_virus[2, ] == 0) 
sum(s4_virus[3, ] == 0) 
# Thus all the cells have viral reads

# add a colume into bc_info dataframe to represent the percentage of cells with 
# corresponding DI boundaries in cells with viral reads
PB2_filtered_bc_info$cell_percentage <- 100 * PB2_filtered_bc_info$cell_number / ncol(s4)
PB1_filtered_bc_info$cell_percentage <- 100 * PB1_filtered_bc_info$cell_number / ncol(s4)
PA_filtered_bc_info$cell_percentage <- 100 * PA_filtered_bc_info$cell_number / ncol(s4)

# Adjust the y-axis to the median of the DI ratios for corresponding DIs by three methods ##
# 1) by the UMI counts including full and gapped reads at 3' peak region
peak_counts <- read.csv("../../../A549_24h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                        row.names=1)
rownames(peak_counts) <- c("PB2", "PB1", "PA")
peak_counts <- peak_counts[, colnames(s4)]

# 2) by the total UMI counts for each viral segment
s4_poly <- s4_virus[1:3, ]


# PB2
# "3p": fall into 3' peak range
# "total": total UMI counts for a given segment
PB2_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s4_poly[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB2_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB2_filtered_bc_info <- merge(PB2_filtered_bc_info, PB2_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_3p)
PB2_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_total)
PB2_filtered_bc_info$grouped_boundary_comb <- as.character(PB2_filtered_bc_info$grouped_boundary_comb)
PB2_filtered_bc_info$boundary_idx <- as.factor(PB2_filtered_bc_info$boundary_idx)
PB2_filtered_bc_info$Color <- as.character(PB2_filtered_bc_info$Color)

write.csv(PB2_filtered_bc_info, file="../../../A549_24h/A549_24hpi_PB2_DI_boundary_ratio_master_info.csv")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")


# PB1
PB1_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s4_poly[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB1_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB1_filtered_bc_info <- merge(PB1_filtered_bc_info, PB1_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_3p)
PB1_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_total)
PB1_filtered_bc_info$grouped_boundary_comb <- as.character(PB1_filtered_bc_info$grouped_boundary_comb)
PB1_filtered_bc_info$boundary_idx <- as.factor(PB1_filtered_bc_info$boundary_idx)
PB1_filtered_bc_info$Color <- as.character(PB1_filtered_bc_info$Color)

write.csv(PB1_filtered_bc_info, file="../../../A549_24h/A549_24hpi_PB1_DI_boundary_ratio_master_info.csv")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

# PA
PA_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                       "median_DI_ratio_3p"=numeric(),
                                       "median_DI_ratio_total"=numeric(),
                                       stringsAsFactors=FALSE)
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == 
                                grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s4_poly[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PA_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                     i_DIratio_total_median)
}
PA_filtered_bc_info <- merge(PA_filtered_bc_info, PA_eachDI_eachcell_ratio, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PA_filtered_bc_info$median_DI_ratio_3p)
PA_filtered_bc_info$median_DI_ratio_total <- as.numeric(PA_filtered_bc_info$median_DI_ratio_total)
PA_filtered_bc_info$grouped_boundary_comb <- as.character(PA_filtered_bc_info$grouped_boundary_comb)
PA_filtered_bc_info$boundary_idx <- as.factor(PA_filtered_bc_info$boundary_idx)
PA_filtered_bc_info$Color <- as.character(PA_filtered_bc_info$Color)

write.csv(PA_filtered_bc_info, file="../../../A549_24h/A549_24hpi_PA_DI_boundary_ratio_master_info.csv")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")



### HBEpC 6hpi ###
## Load data
boundary_criteria <- read.csv("../Filtered_Grouped_DI_boundaries_HBEpC_6h.csv")

## Filter the single cell data
PB2_boundary_comb <- HBEpC_6h_sc_PB2_boundary_df
PB2_boundary_comb$boundary_comb <- paste(PB2_boundary_comb$start, 
                                         PB2_boundary_comb$stop, sep="^")

PB1_boundary_comb <- HBEpC_6h_sc_PB1_boundary_df
PB1_boundary_comb$boundary_comb <- paste(PB1_boundary_comb$start, 
                                         PB1_boundary_comb$stop, sep="^")

PA_boundary_comb <- HBEpC_6h_sc_PA_boundary_df
PA_boundary_comb$boundary_comb <- paste(PA_boundary_comb$start, 
                                        PA_boundary_comb$stop, sep="^")

PB2_boundary_comb_filtered <- merge(PB2_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB2", ])
PB1_boundary_comb_filtered <- merge(PB1_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB1", ])
PA_boundary_comb_filtered <- merge(PA_boundary_comb, 
                                   boundary_criteria[boundary_criteria$seg == "PA", ])

## Extract the information about the frequency of a boundary combination in the dataset
PB2_filtered_freq <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PB1_filtered_freq <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PA_filtered_freq <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                          "grouped_boundary_comb")
colnames(PB2_filtered_freq)[2] <- "counts"
colnames(PB1_filtered_freq)[2] <- "counts"
colnames(PA_filtered_freq)[2] <- "counts"

## Extraction the information about the number of cells in which a boundary was observed
PB2_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB2_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB2_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB2", -1])
PB2_filtered_bc_info <- merge(PB2_filtered_freq, boundary_criteria_PB2_unique, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info <- merge(PB2_filtered_cfreq, PB2_filtered_bc_info, 
                              by="boundary_idx")
# Note: "counts" represents the number of times a boundary observed in the whole dataset (all the cells)
# "cell_number" represents the number of cells in which a boundary was observed
PB2_filtered_bc_info$Overlap_stock <- factor(PB2_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PB1_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB1_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB1_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB1", -1])
PB1_filtered_bc_info <- merge(PB1_filtered_freq, boundary_criteria_PB1_unique, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info <- merge(PB1_filtered_cfreq, PB1_filtered_bc_info, 
                              by="boundary_idx")
PB1_filtered_bc_info$Overlap_stock <- factor(PB1_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PA_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                "cell_number"=integer())
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PA_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PA_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PA", -1])
PA_filtered_bc_info <- merge(PA_filtered_freq, boundary_criteria_PA_unique, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info <- merge(PA_filtered_cfreq, PA_filtered_bc_info, 
                             by="boundary_idx")
PA_filtered_bc_info$Overlap_stock <- factor(PA_filtered_bc_info$Overlap_stock,
                                            levels=c("Yes", "No"))

## Count the filtered DIs in each cell
PB2_filtered_count <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PB1_filtered_count <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PA_filtered_count <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                           "cell")
# merge above data into a dataframe
Poly_filtered_count <- merge(PB2_filtered_count, PB1_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count <- merge(Poly_filtered_count, PA_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count[is.na(Poly_filtered_count)] <- 0
colnames(Poly_filtered_count) <- c("cell", "PB2_DI_count", "PB1_DI_count", "PA_DI_count")
# export
write.csv(Poly_filtered_count, 
          file="../../../HBEpC_6h/HBEpC_6h_polymerases_DI_filtered-grouped_counts.csv")


## Adjust the x and y axis and re-plot ##
# Adjust the x-aixs to percentage #
s6_virus <- s6[(nrow(s6)-7):nrow(s6), ]
s6_vsum <- as.data.frame(colSums(s6_virus)) 
s6_host <- s6[1:(nrow(s6)-8), ]

# Check if any cell with PB2/PB1/PA undetected
sum(s6_virus[1, ] == 0) 
sum(s6_virus[2, ] == 0) 
sum(s6_virus[3, ] == 0) 
# Thus some cells do not have viral reads for some segments

# add a colume into bc_info dataframe to represent the percentage of cells with 
# corresponding DI boundaries in cells with viral reads for corresponding segments
PB2_filtered_bc_info$cell_percentage <- 
  100 * PB2_filtered_bc_info$cell_number / (ncol(s6)-sum(s6_virus[1, ] == 0))
PB1_filtered_bc_info$cell_percentage <- 
  100 * PB1_filtered_bc_info$cell_number / (ncol(s6)-sum(s6_virus[2, ] == 0))
PA_filtered_bc_info$cell_percentage <- 
  100 * PA_filtered_bc_info$cell_number / (ncol(s6)-sum(s6_virus[3, ] == 0))

# Adjust the y-axis to the median of the DI ratios for corresponding DIs by three methods ##
# 1) by the UMI counts including full and gapped reads at 3' peak region
peak_counts <- read.csv("../../../HBEpC_6h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                        row.names=1)
rownames(peak_counts) <- c("PB2", "PB1", "PA")
peak_counts <- peak_counts[, colnames(s6)]

# 2) by the total UMI counts for each viral segment
s6_poly <- s6_virus[1:3, ]


# PB2
# "3p": fall into 3' peak range
# "total": total UMI counts for a given segment
PB2_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s6_poly[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_raw <- i_subset_freqtb_nonzero$DI_ratio_3p
  i_DIratio_3p_raw <- i_DIratio_3p_raw[!is.infinite(i_DIratio_3p_raw)]
  i_DIratio_3p_median <- median(i_DIratio_3p_raw, na.rm=TRUE)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB2_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB2_filtered_bc_info <- merge(PB2_filtered_bc_info, PB2_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_3p)
PB2_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_total)
PB2_filtered_bc_info$grouped_boundary_comb <- as.character(PB2_filtered_bc_info$grouped_boundary_comb)
PB2_filtered_bc_info$boundary_idx <- as.factor(PB2_filtered_bc_info$boundary_idx)
PB2_filtered_bc_info$Color <- as.character(PB2_filtered_bc_info$Color)

write.csv(PB2_filtered_bc_info, file="../../../HBEpC_6h/HBEpC_6hpi_PB2_DI_boundary_ratio_master_info.csv")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")


# PB1
PB1_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s6_poly[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_raw <- i_subset_freqtb_nonzero$DI_ratio_3p
  i_DIratio_3p_raw <- i_DIratio_3p_raw[!is.infinite(i_DIratio_3p_raw)]
  i_DIratio_3p_median <- median(i_DIratio_3p_raw, na.rm=TRUE)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB1_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB1_filtered_bc_info <- merge(PB1_filtered_bc_info, PB1_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_3p)
PB1_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_total)
PB1_filtered_bc_info$grouped_boundary_comb <- as.character(PB1_filtered_bc_info$grouped_boundary_comb)
PB1_filtered_bc_info$boundary_idx <- as.factor(PB1_filtered_bc_info$boundary_idx)
PB1_filtered_bc_info$Color <- as.character(PB1_filtered_bc_info$Color)

write.csv(PB1_filtered_bc_info, file="../../../HBEpC_6h/HBEpC_6hpi_PB1_DI_boundary_ratio_master_info.csv")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

# PA
PA_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                       "median_DI_ratio_3p"=numeric(),
                                       "median_DI_ratio_total"=numeric(),
                                       stringsAsFactors=FALSE)
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == 
                                grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s6_poly[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_raw <- i_subset_freqtb_nonzero$DI_ratio_3p
  i_DIratio_3p_raw <- i_DIratio_3p_raw[!is.infinite(i_DIratio_3p_raw)]
  i_DIratio_3p_median <- median(i_DIratio_3p_raw, na.rm=TRUE)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PA_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                     i_DIratio_total_median)
}
PA_filtered_bc_info <- merge(PA_filtered_bc_info, PA_eachDI_eachcell_ratio, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PA_filtered_bc_info$median_DI_ratio_3p)
PA_filtered_bc_info$median_DI_ratio_total <- as.numeric(PA_filtered_bc_info$median_DI_ratio_total)
PA_filtered_bc_info$grouped_boundary_comb <- as.character(PA_filtered_bc_info$grouped_boundary_comb)
PA_filtered_bc_info$boundary_idx <- as.factor(PA_filtered_bc_info$boundary_idx)
PA_filtered_bc_info$Color <- as.character(PA_filtered_bc_info$Color)

write.csv(PA_filtered_bc_info, file="../../../HBEpC_6h/HBEpC_6hpi_PA_DI_boundary_ratio_master_info.csv")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")



### HBEpC 12hpi ###
## Load data
boundary_criteria <- read.csv("../Filtered_Grouped_DI_boundaries_HBEpC_12h.csv")

## Filter the single cell data
PB2_boundary_comb <- HBEpC_12h_sc_PB2_boundary_df
PB2_boundary_comb$boundary_comb <- paste(PB2_boundary_comb$start, 
                                         PB2_boundary_comb$stop, sep="^")

PB1_boundary_comb <- HBEpC_12h_sc_PB1_boundary_df
PB1_boundary_comb$boundary_comb <- paste(PB1_boundary_comb$start, 
                                         PB1_boundary_comb$stop, sep="^")

PA_boundary_comb <- HBEpC_12h_sc_PA_boundary_df
PA_boundary_comb$boundary_comb <- paste(PA_boundary_comb$start, 
                                        PA_boundary_comb$stop, sep="^")

PB2_boundary_comb_filtered <- merge(PB2_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB2", ])
PB1_boundary_comb_filtered <- merge(PB1_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB1", ])
PA_boundary_comb_filtered <- merge(PA_boundary_comb, 
                                   boundary_criteria[boundary_criteria$seg == "PA", ])

## Extract the information about the frequency of a boundary combination in the dataset
PB2_filtered_freq <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PB1_filtered_freq <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PA_filtered_freq <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                          "grouped_boundary_comb")
colnames(PB2_filtered_freq)[2] <- "counts"
colnames(PB1_filtered_freq)[2] <- "counts"
colnames(PA_filtered_freq)[2] <- "counts"

## Extraction the information about the number of cells in which a boundary was observed
PB2_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB2_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB2_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB2", -1])
PB2_filtered_bc_info <- merge(PB2_filtered_freq, boundary_criteria_PB2_unique, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info <- merge(PB2_filtered_cfreq, PB2_filtered_bc_info, 
                              by="boundary_idx")
# Note: "counts" represents the number of times a boundary observed in the whole dataset (all the cells)
# "cell_number" represents the number of cells in which a boundary was observed
PB2_filtered_bc_info$Overlap_stock <- factor(PB2_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PB1_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB1_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB1_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB1", -1])
PB1_filtered_bc_info <- merge(PB1_filtered_freq, boundary_criteria_PB1_unique, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info <- merge(PB1_filtered_cfreq, PB1_filtered_bc_info, 
                              by="boundary_idx")
PB1_filtered_bc_info$Overlap_stock <- factor(PB1_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PA_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                "cell_number"=integer())
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PA_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PA_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PA", -1])
PA_filtered_bc_info <- merge(PA_filtered_freq, boundary_criteria_PA_unique, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info <- merge(PA_filtered_cfreq, PA_filtered_bc_info, 
                             by="boundary_idx")
PA_filtered_bc_info$Overlap_stock <- factor(PA_filtered_bc_info$Overlap_stock,
                                            levels=c("Yes", "No"))

## Count the filtered DIs in each cell
PB2_filtered_count <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PB1_filtered_count <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PA_filtered_count <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                           "cell")
# merge above data into a dataframe
Poly_filtered_count <- merge(PB2_filtered_count, PB1_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count <- merge(Poly_filtered_count, PA_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count[is.na(Poly_filtered_count)] <- 0
colnames(Poly_filtered_count) <- c("cell", "PB2_DI_count", "PB1_DI_count", "PA_DI_count")
# export
write.csv(Poly_filtered_count, 
          file="../../../HBEpC_12h/HBEpC_12h_polymerases_DI_filtered-grouped_counts.csv")


## Adjust the x and y axis and re-plot ##
# Adjust the x-aixs to percentage #
s7_virus <- s7[(nrow(s7)-7):nrow(s7), ]
s7_vsum <- as.data.frame(colSums(s7_virus)) 
s7_host <- s7[1:(nrow(s7)-8), ]

# Check if any cell with PB2/PB1/PA undetected
sum(s7_virus[1, ] == 0) 
sum(s7_virus[2, ] == 0) 
sum(s7_virus[3, ] == 0) 
# Thus all the cells have viral reads

# add a colume into bc_info dataframe to represent the percentage of cells with 
# corresponding DI boundaries in cells with viral reads
PB2_filtered_bc_info$cell_percentage <- 100 * PB2_filtered_bc_info$cell_number / ncol(s7)
PB1_filtered_bc_info$cell_percentage <- 100 * PB1_filtered_bc_info$cell_number / ncol(s7)
PA_filtered_bc_info$cell_percentage <- 100 * PA_filtered_bc_info$cell_number / ncol(s7)

# Adjust the y-axis to the median of the DI ratios for corresponding DIs by three methods ##
# 1) by the UMI counts including full and gapped reads at 3' peak region
peak_counts <- read.csv("../../../HBEpC_12h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                        row.names=1)
rownames(peak_counts) <- c("PB2", "PB1", "PA")
peak_counts <- peak_counts[, colnames(s7)]

# 2) by the total UMI counts for each viral segment
s7_poly <- s7_virus[1:3, ]


# PB2
# "3p": fall into 3' peak range
# "total": total UMI counts for a given segment
PB2_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s7_poly[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB2_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB2_filtered_bc_info <- merge(PB2_filtered_bc_info, PB2_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_3p)
PB2_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_total)
PB2_filtered_bc_info$grouped_boundary_comb <- as.character(PB2_filtered_bc_info$grouped_boundary_comb)
PB2_filtered_bc_info$boundary_idx <- as.factor(PB2_filtered_bc_info$boundary_idx)
PB2_filtered_bc_info$Color <- as.character(PB2_filtered_bc_info$Color)

write.csv(PB2_filtered_bc_info, file="../../../HBEpC_12h/HBEpC_12hpi_PB2_DI_boundary_ratio_master_info.csv")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")


# PB1
PB1_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s7_poly[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB1_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB1_filtered_bc_info <- merge(PB1_filtered_bc_info, PB1_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_3p)
PB1_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_total)
PB1_filtered_bc_info$grouped_boundary_comb <- as.character(PB1_filtered_bc_info$grouped_boundary_comb)
PB1_filtered_bc_info$boundary_idx <- as.factor(PB1_filtered_bc_info$boundary_idx)
PB1_filtered_bc_info$Color <- as.character(PB1_filtered_bc_info$Color)

write.csv(PB1_filtered_bc_info, file="../../../HBEpC_12h/HBEpC_12hpi_PB1_DI_boundary_ratio_master_info.csv")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")


# PA
PA_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                       "median_DI_ratio_3p"=numeric(),
                                       "median_DI_ratio_total"=numeric(),
                                       stringsAsFactors=FALSE)
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == 
                                grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s7_poly[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PA_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                     i_DIratio_total_median)
}
PA_filtered_bc_info <- merge(PA_filtered_bc_info, PA_eachDI_eachcell_ratio, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PA_filtered_bc_info$median_DI_ratio_3p)
PA_filtered_bc_info$median_DI_ratio_total <- as.numeric(PA_filtered_bc_info$median_DI_ratio_total)
PA_filtered_bc_info$grouped_boundary_comb <- as.character(PA_filtered_bc_info$grouped_boundary_comb)
PA_filtered_bc_info$boundary_idx <- as.factor(PA_filtered_bc_info$boundary_idx)
PA_filtered_bc_info$Color <- as.character(PA_filtered_bc_info$Color)

write.csv(PA_filtered_bc_info, file="../../../HBEpC_12h/HBEpC_12hpi_PA_DI_boundary_ratio_master_info.csv")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")



### HBEpC 24hpi ###
## Load data
boundary_criteria <- read.csv("../Filtered_Grouped_DI_boundaries_HBEpC_24h.csv")

## Filter the single cell data
PB2_boundary_comb <- HBEpC_24h_sc_PB2_boundary_df
PB2_boundary_comb$boundary_comb <- paste(PB2_boundary_comb$start, 
                                         PB2_boundary_comb$stop, sep="^")

PB1_boundary_comb <- HBEpC_24h_sc_PB1_boundary_df
PB1_boundary_comb$boundary_comb <- paste(PB1_boundary_comb$start, 
                                         PB1_boundary_comb$stop, sep="^")

PA_boundary_comb <- HBEpC_24h_sc_PA_boundary_df
PA_boundary_comb$boundary_comb <- paste(PA_boundary_comb$start, 
                                        PA_boundary_comb$stop, sep="^")

PB2_boundary_comb_filtered <- merge(PB2_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB2", ])
PB1_boundary_comb_filtered <- merge(PB1_boundary_comb, 
                                    boundary_criteria[boundary_criteria$seg == "PB1", ])
PA_boundary_comb_filtered <- merge(PA_boundary_comb, 
                                   boundary_criteria[boundary_criteria$seg == "PA", ])

## Extract the information about the frequency of a boundary combination in the dataset
PB2_filtered_freq <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PB1_filtered_freq <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                           "grouped_boundary_comb")
PA_filtered_freq <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")], 
                          "grouped_boundary_comb")
colnames(PB2_filtered_freq)[2] <- "counts"
colnames(PB1_filtered_freq)[2] <- "counts"
colnames(PA_filtered_freq)[2] <- "counts"

## Extraction the information about the number of cells in which a boundary was observed
PB2_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB2_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB2_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB2", -1])
PB2_filtered_bc_info <- merge(PB2_filtered_freq, boundary_criteria_PB2_unique, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info <- merge(PB2_filtered_cfreq, PB2_filtered_bc_info, 
                              by="boundary_idx")
# Note: "counts" represents the number of times a boundary observed in the whole dataset (all the cells)
# "cell_number" represents the number of cells in which a boundary was observed
PB2_filtered_bc_info$Overlap_stock <- factor(PB2_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PB1_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                 "cell_number"=integer())
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PB1_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PB1_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PB1", -1])
PB1_filtered_bc_info <- merge(PB1_filtered_freq, boundary_criteria_PB1_unique, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info <- merge(PB1_filtered_cfreq, PB1_filtered_bc_info, 
                              by="boundary_idx")
PB1_filtered_bc_info$Overlap_stock <- factor(PB1_filtered_bc_info$Overlap_stock,
                                             levels=c("Yes", "No"))

PA_filtered_cfreq <- data.frame("boundary_idx"=integer(),
                                "cell_number"=integer())
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  boundary <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  boundary_subdf <- PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == boundary, ]
  boundary_idx <- as.integer(as.character(unique(boundary_subdf$boundary_idx)))
  ccount <- length(unique(boundary_subdf$cell))
  PA_filtered_cfreq[i, ] <- data.frame(boundary_idx, ccount)
}
boundary_criteria_PA_unique <- 
  unique(boundary_criteria[boundary_criteria$seg == "PA", -1])
PA_filtered_bc_info <- merge(PA_filtered_freq, boundary_criteria_PA_unique, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info <- merge(PA_filtered_cfreq, PA_filtered_bc_info, 
                             by="boundary_idx")
PA_filtered_bc_info$Overlap_stock <- factor(PA_filtered_bc_info$Overlap_stock,
                                            levels=c("Yes", "No"))

## Count the filtered DIs in each cell
PB2_filtered_count <- count(PB2_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PB1_filtered_count <- count(PB1_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                            "cell")
PA_filtered_count <- count(PA_boundary_comb_filtered[, c("cell", "grouped_boundary_comb")],
                           "cell")
# merge above data into a dataframe
Poly_filtered_count <- merge(PB2_filtered_count, PB1_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count <- merge(Poly_filtered_count, PA_filtered_count, by="cell",
                             all=TRUE)
Poly_filtered_count[is.na(Poly_filtered_count)] <- 0
colnames(Poly_filtered_count) <- c("cell", "PB2_DI_count", "PB1_DI_count", "PA_DI_count")
# export
write.csv(Poly_filtered_count, 
          file="../../../HBEpC_24h/HBEpC_24h_polymerases_DI_filtered-grouped_counts.csv")


## Adjust the x and y axis and re-plot ##
# Adjust the x-aixs to percentage #
s8_virus <- s8[(nrow(s8)-7):nrow(s8), ]
s8_vsum <- as.data.frame(colSums(s8_virus)) 
s8_host <- s8[1:(nrow(s8)-8), ]

# Check if any cell with PB2/PB1/PA undetected
sum(s8_virus[1, ] == 0) 
sum(s8_virus[2, ] == 0) 
sum(s8_virus[3, ] == 0) 
# Thus all the cells have viral reads

# add a colume into bc_info dataframe to represent the percentage of cells with 
# corresponding DI boundaries in cells with viral reads
PB2_filtered_bc_info$cell_percentage <- 100 * PB2_filtered_bc_info$cell_number / ncol(s8)
PB1_filtered_bc_info$cell_percentage <- 100 * PB1_filtered_bc_info$cell_number / ncol(s8)
PA_filtered_bc_info$cell_percentage <- 100 * PA_filtered_bc_info$cell_number / ncol(s8)

# Adjust the y-axis to the median of the DI ratios for corresponding DIs by three methods ##
# 1) by the UMI counts including full and gapped reads at 3' peak region
peak_counts <- read.csv("../../../HBEpC_24h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                        row.names=1)
rownames(peak_counts) <- c("PB2", "PB1", "PA")
peak_counts <- peak_counts[, colnames(s8)]

# 2) by the total UMI counts for each viral segment
s8_poly <- s8_virus[1:3, ]


# PB2
# "3p": fall into 3' peak range
# "total": total UMI counts for a given segment
PB2_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB2_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB2_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB2_boundary_comb_filtered[PB2_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s8_poly[1, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB2_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB2_filtered_bc_info <- merge(PB2_filtered_bc_info, PB2_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB2_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_3p)
PB2_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB2_filtered_bc_info$median_DI_ratio_total)
PB2_filtered_bc_info$grouped_boundary_comb <- as.character(PB2_filtered_bc_info$grouped_boundary_comb)
PB2_filtered_bc_info$boundary_idx <- as.factor(PB2_filtered_bc_info$boundary_idx)
PB2_filtered_bc_info$Color <- as.character(PB2_filtered_bc_info$Color)

write.csv(PB2_filtered_bc_info, file="../../../HBEpC_24h/HBEpC_24hpi_PB2_DI_boundary_ratio_master_info.csv")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB2_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB2_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")


# PB1
PB1_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                        "median_DI_ratio_3p"=numeric(),
                                        "median_DI_ratio_total"=numeric(),
                                        stringsAsFactors=FALSE)
for(i in 1:length(unique(PB1_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PB1_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PB1_boundary_comb_filtered[PB1_boundary_comb_filtered$grouped_boundary_comb == 
                                 grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s8_poly[2, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PB1_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                      i_DIratio_total_median)
}
PB1_filtered_bc_info <- merge(PB1_filtered_bc_info, PB1_eachDI_eachcell_ratio, 
                              by="grouped_boundary_comb")
PB1_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_3p)
PB1_filtered_bc_info$median_DI_ratio_total <- as.numeric(PB1_filtered_bc_info$median_DI_ratio_total)
PB1_filtered_bc_info$grouped_boundary_comb <- as.character(PB1_filtered_bc_info$grouped_boundary_comb)
PB1_filtered_bc_info$boundary_idx <- as.factor(PB1_filtered_bc_info$boundary_idx)
PB1_filtered_bc_info$Color <- as.character(PB1_filtered_bc_info$Color)

write.csv(PB1_filtered_bc_info, file="../../../HBEpC_24h/HBEpC_24hpi_PB1_DI_boundary_ratio_master_info.csv")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PB1_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PB1_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")


# PA
PA_eachDI_eachcell_ratio <- data.frame("grouped_boundary_comb"=character(),
                                       "median_DI_ratio_3p"=numeric(),
                                       "median_DI_ratio_total"=numeric(),
                                       stringsAsFactors=FALSE)
for(i in 1:length(unique(PA_boundary_comb_filtered$grouped_boundary_comb))) {
  grouped_boundary_comb <- as.character(unique(PA_boundary_comb_filtered$grouped_boundary_comb)[i])
  i_subset <- 
    PA_boundary_comb_filtered[PA_boundary_comb_filtered$grouped_boundary_comb == 
                                grouped_boundary_comb, ]
  i_subset_freqtb <- as.data.frame(table(i_subset[, c("cell", "grouped_boundary_comb")]))
  i_subset_freqtb_nonzero <- i_subset_freqtb[i_subset_freqtb$Freq > 0, ]
  i_subset_freqtb_nonzero$cell <- as.character(i_subset_freqtb_nonzero$cell)
  i_subset_freqtb_nonzero$peak_count <- t(peak_counts[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$total_count <- t(s8_poly[3, i_subset_freqtb_nonzero$cell])
  i_subset_freqtb_nonzero$DI_ratio_3p <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$peak_count
  i_subset_freqtb_nonzero$DI_ratio_total <- 
    i_subset_freqtb_nonzero$Freq / i_subset_freqtb_nonzero$total_count
  i_DIratio_3p_median <- median(i_subset_freqtb_nonzero$DI_ratio_3p)
  i_DIratio_total_median <- median(i_subset_freqtb_nonzero$DI_ratio_total)
  PA_eachDI_eachcell_ratio[i, ] <- c(grouped_boundary_comb, i_DIratio_3p_median,
                                     i_DIratio_total_median)
}
PA_filtered_bc_info <- merge(PA_filtered_bc_info, PA_eachDI_eachcell_ratio, 
                             by="grouped_boundary_comb")
PA_filtered_bc_info$median_DI_ratio_3p <- as.numeric(PA_filtered_bc_info$median_DI_ratio_3p)
PA_filtered_bc_info$median_DI_ratio_total <- as.numeric(PA_filtered_bc_info$median_DI_ratio_total)
PA_filtered_bc_info$grouped_boundary_comb <- as.character(PA_filtered_bc_info$grouped_boundary_comb)
PA_filtered_bc_info$boundary_idx <- as.factor(PA_filtered_bc_info$boundary_idx)
PA_filtered_bc_info$Color <- as.character(PA_filtered_bc_info$Color)

write.csv(PA_filtered_bc_info, file="../../../HBEpC_24h/HBEpC_24hpi_PA_DI_boundary_ratio_master_info.csv")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_3p, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

ggplot(PA_filtered_bc_info, aes(x=cell_percentage, y=median_DI_ratio_total, color=grouped_boundary_comb)) +
  geom_point(aes(shape=Overlap_stock), size=5) +
  scale_color_manual(values=PA_filtered_bc_info$Color) +
  scale_y_continuous(breaks=seq(0, 1, 0.1), limits=c(0, 1)) +
  scale_x_continuous(breaks=seq(0, 100, 25), limits=c(0, 100)) +
  theme(panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        text=element_text(size=15),
        legend.background=element_rect(color="white")) +
  labs(x="Cell percentages", y="DI ratios", color="Grouped boundaries", shape="From stock")

```



#### Calculate the overall relative abundance of all the defective viral transcripts per cell and export it ####
```{r warning=FALSE, message=FALSE, include=TRUE, eval=FALSE}
### A549 6hpi ###
setwd("~/Documents/10XGenomics_BulkRNAseq_qPCR_2017Fall/10X_data_July2018_v2/")

## Load data
DI_count <- read.csv("A549_6h/A549_6h_polymerases_DI_filtered-grouped_counts.csv",
                     row.names=1)
peak_count <- read.csv("A549_6h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                       row.names=1)
peak_count <- as.data.frame(t(peak_count))
colnames(peak_count) <- c("PB2", "PB1", "PA")
load("CellQC/A549/A549ALLsamples_QCed_gene-expression-matrix_S2_A549-6h.RData")
seg_count <- s2[(nrow(s2)-7):(nrow(s2)-5), ]
seg_count <- as.data.frame(t(seg_count))

peak_count$cell <- rownames(peak_count)
seg_count$cell <- rownames(seg_count)
DI_info_master <- merge(DI_count, seg_count, by="cell")
DI_info_master <- merge(DI_info_master, peak_count, by="cell")
colnames(DI_info_master)[5:10] <- c("PB2_seg_count", "PB1_seg_count", "PA_seg_count",
                                    "PB2_peak_count", "PB1_peak_count", "PA_peak_count")

## Calculate DI ratio 
DI_info_master$PB2_DI_ratio_total <- DI_info_master$PB2_DI_count / DI_info_master$PB2_seg_count
DI_info_master$PB1_DI_ratio_total <- DI_info_master$PB1_DI_count / DI_info_master$PB1_seg_count
DI_info_master$PA_DI_ratio_total <- DI_info_master$PA_DI_count / DI_info_master$PA_seg_count

DI_info_master$PB2_DI_ratio_3p <- DI_info_master$PB2_DI_count / DI_info_master$PB2_peak_count
DI_info_master$PB1_DI_ratio_3p <- DI_info_master$PB1_DI_count / DI_info_master$PB1_peak_count
DI_info_master$PA_DI_ratio_3p <- DI_info_master$PA_DI_count / DI_info_master$PA_peak_count

# Export 
write.csv(DI_info_master, file="A549_6h/A549_6h_polymerases_DI-ratio.csv")


## Check the distribution of DI ratios
DI_ratio_total <- DI_info_master[, c(1, 11:13)]
DI_ratio_total_melt <- melt(DI_ratio_total, id.vars="cell")
DI_ratio_total_melt$variable <- gsub("_DI_ratio_total", "", DI_ratio_total_melt$variable)
DI_ratio_total_melt$variable <- factor(DI_ratio_total_melt$variable, 
                                       levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_total_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 0.8, 0.1)) +
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")

DI_ratio_3p <- DI_info_master[, c(1, 14:16)]
DI_ratio_3p_melt <- melt(DI_ratio_3p, id.vars="cell")
DI_ratio_3p_melt$variable <- gsub("_DI_ratio_3p", "", DI_ratio_3p_melt$variable)
DI_ratio_3p_melt$variable <- factor(DI_ratio_3p_melt$variable, levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_3p_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 4, 0.5)) +
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")



### A549 12hpi ###
## Load data
DI_count <- read.csv("A549_12h/A549_12h_polymerases_DI_filtered-grouped_counts.csv",
                     row.names=1)
peak_count <- read.csv("A549_12h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                       row.names=1)
peak_count <- as.data.frame(t(peak_count))
colnames(peak_count) <- c("PB2", "PB1", "PA")
load("CellQC/A549/A549ALLsamples_QCed_gene-expression-matrix_S3_A549-12h.RData")
seg_count <- s3[((nrow(s3)-7):(nrow(s3)-5)), ]
seg_count <- as.data.frame(t(seg_count))

peak_count$cell <- rownames(peak_count)
seg_count$cell <- rownames(seg_count)
DI_info_master <- merge(DI_count, seg_count, by="cell")
DI_info_master <- merge(DI_info_master, peak_count, by="cell")
colnames(DI_info_master)[5:10] <- c("PB2_seg_count", "PB1_seg_count", "PA_seg_count",
                                    "PB2_peak_count", "PB1_peak_count", "PA_peak_count")

## Calculate DI ratio 
DI_info_master$PB2_DI_ratio_total <- DI_info_master$PB2_DI_count / DI_info_master$PB2_seg_count
DI_info_master$PB1_DI_ratio_total <- DI_info_master$PB1_DI_count / DI_info_master$PB1_seg_count
DI_info_master$PA_DI_ratio_total <- DI_info_master$PA_DI_count / DI_info_master$PA_seg_count

DI_info_master$PB2_DI_ratio_3p <- DI_info_master$PB2_DI_count / DI_info_master$PB2_peak_count
DI_info_master$PB1_DI_ratio_3p <- DI_info_master$PB1_DI_count / DI_info_master$PB1_peak_count
DI_info_master$PA_DI_ratio_3p <- DI_info_master$PA_DI_count / DI_info_master$PA_peak_count

# Export 
write.csv(DI_info_master, file="A549_12h/A549_12h_polymerases_DI-ratio.csv")


## Check the distribution of DI ratios
DI_ratio_total <- DI_info_master[, c(1, 11:13)]
DI_ratio_total_melt <- melt(DI_ratio_total, id.vars="cell")
DI_ratio_total_melt$variable <- gsub("_DI_ratio_total", "", DI_ratio_total_melt$variable)
DI_ratio_total_melt$variable <- factor(DI_ratio_total_melt$variable, 
                                       levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_total_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 0.8, 0.1)) +
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")

DI_ratio_3p <- DI_info_master[, c(1, 14:16)]
DI_ratio_3p_melt <- melt(DI_ratio_3p, id.vars="cell")
DI_ratio_3p_melt$variable <- gsub("_DI_ratio_3p", "", DI_ratio_3p_melt$variable)
DI_ratio_3p_melt$variable <- factor(DI_ratio_3p_melt$variable, levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_3p_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 4, 0.5)) +
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")



### A549 24hpi ###
## Load data
DI_count <- read.csv("A549_24h/A549_24h_polymerases_DI_filtered-grouped_counts.csv",
                     row.names=1)
peak_count <- read.csv("A549_24h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                       row.names=1)
peak_count <- as.data.frame(t(peak_count))
colnames(peak_count) <- c("PB2", "PB1", "PA")
load("CellQC/A549/A549ALLsamples_QCed_gene-expression-matrix_S4_A549-24h.RData")
seg_count <- s4[((nrow(s4)-7):(nrow(s4)-5)), ]
seg_count <- as.data.frame(t(seg_count))

peak_count$cell <- rownames(peak_count)
seg_count$cell <- rownames(seg_count)
DI_info_master <- merge(DI_count, seg_count, by="cell")
DI_info_master <- merge(DI_info_master, peak_count, by="cell")
colnames(DI_info_master)[5:10] <- c("PB2_seg_count", "PB1_seg_count", "PA_seg_count",
                                    "PB2_peak_count", "PB1_peak_count", "PA_peak_count")

## Calculate DI ratio 
DI_info_master$PB2_DI_ratio_total <- DI_info_master$PB2_DI_count / DI_info_master$PB2_seg_count
DI_info_master$PB1_DI_ratio_total <- DI_info_master$PB1_DI_count / DI_info_master$PB1_seg_count
DI_info_master$PA_DI_ratio_total <- DI_info_master$PA_DI_count / DI_info_master$PA_seg_count

DI_info_master$PB2_DI_ratio_3p <- DI_info_master$PB2_DI_count / DI_info_master$PB2_peak_count
DI_info_master$PB1_DI_ratio_3p <- DI_info_master$PB1_DI_count / DI_info_master$PB1_peak_count
DI_info_master$PA_DI_ratio_3p <- DI_info_master$PA_DI_count / DI_info_master$PA_peak_count

# Export 
write.csv(DI_info_master, file="A549_24h/A549_24h_polymerases_DI-ratio.csv")


## Check the distribution of DI ratios
DI_ratio_total <- DI_info_master[, c(1, 11:13)]
DI_ratio_total_melt <- melt(DI_ratio_total, id.vars="cell")
DI_ratio_total_melt$variable <- gsub("_DI_ratio_total", "", DI_ratio_total_melt$variable)
DI_ratio_total_melt$variable <- factor(DI_ratio_total_melt$variable, 
                                       levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_total_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 0.8, 0.1)) +
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")

DI_ratio_3p <- DI_info_master[, c(1, 14:16)]
DI_ratio_3p_melt <- melt(DI_ratio_3p, id.vars="cell")
DI_ratio_3p_melt$variable <- gsub("_DI_ratio_3p", "", DI_ratio_3p_melt$variable)
DI_ratio_3p_melt$variable <- factor(DI_ratio_3p_melt$variable, levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_3p_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 3, 0.5)) +
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")



### All three A549 time points together ###
## Load data
A549_6h_DI <- read.csv("A549_6h/A549_6h_polymerases_DI-ratio.csv", row.names=1)
A549_12h_DI <- read.csv("A549_12h/A549_12h_polymerases_DI-ratio.csv", row.names=1)
A549_24h_DI <- read.csv("A549_24h/A549_24h_polymerases_DI-ratio.csv", row.names=1)

## DI normalized to the total segment UMI counts
## Prepare and plot
A549_DI <- rbind(A549_6h_DI[, c(1, 11:13)], A549_12h_DI[, c(1, 11:13)], 
                 A549_24h_DI[, c(1, 11:13)])
A549_DI$sample <- rep(c("6hpi", "12hpi", "24hpi"), 
                      c(nrow(A549_6h_DI), nrow(A549_12h_DI), nrow(A549_24h_DI)))
A549_DI$sample <- factor(A549_DI$sample, levels=c("6hpi", "12hpi", "24hpi"))

A549_DI_melt <- melt(A549_DI, id.vars=c("cell", "sample"))
A549_DI_melt$variable <- gsub("_DI_ratio_total", "", A549_DI_melt$variable)
A549_DI_melt$variable <- factor(A549_DI_melt$variable, levels=c("PB2", "PB1", "PA"))

# raincloud plot
A549_DI_melt$value <- A549_DI_melt$value * 100
ggplot(data=A549_DI_melt, aes(x=sample, y=value, fill=sample)) +
  facet_wrap(~variable) +
  geom_flat_violin(position=position_nudge(x=.2, y=0), alpha=.8) +
  geom_point(aes(y=value, color=sample), position=position_jitter(width=.15), size=.5, alpha=0.8) +
  geom_boxplot(width=.2, guides=FALSE, outlier.shape=NA, alpha=0, lwd=0.8) +
  scale_y_continuous(breaks=seq(0, 80, 10), limits=c(-0.01, 85)) +
  scale_color_manual(values=c("#b1df01", "olivedrab4", "#466b5d")) +
  scale_fill_manual(values=c("#b1df01", "olivedrab4", "#466b5d")) +
  theme(plot.title=element_text(size=15, hjust = 0.5), 
        axis.title=element_text(size=15),
        text=element_text(size=15),
        panel.background=element_rect(fill="white", color="grey50"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  labs(list(title="Distribution of relative abundance of viral transcripts", 
            x="Time point", y="Relative abundance of viral transcripts"))


## Test if the distribution is statistical different
kruskal.test(PB2_DI_ratio_total ~ sample, A549_DI)
kruskal.test(PB1_DI_ratio_total ~ sample, A549_DI)
kruskal.test(PA_DI_ratio_total ~ sample, A549_DI) 

# 6h vs 12h
DI_6hvs12h <- A549_DI[A549_DI$sample == "6hpi" | A549_DI$sample == "12hpi", ]
wilcox.test(PB2_DI_ratio_total ~ sample, DI_6hvs12h)
wilcox.test(PB1_DI_ratio_total ~ sample, DI_6hvs12h)
wilcox.test(PA_DI_ratio_total ~ sample, DI_6hvs12h) 

wilcox.test(PB1_DI_ratio_total ~ sample, DI_6hvs12h, alternative="greater")
wilcox.test(PA_DI_ratio_total ~ sample, DI_6hvs12h, alternative="greater") 

# 6h vs 24h
DI_6hvs24h <- A549_DI[A549_DI$sample == "6hpi" | A549_DI$sample == "24hpi", ]
wilcox.test(PB2_DI_ratio_total ~ sample, DI_6hvs24h)
wilcox.test(PB1_DI_ratio_total ~ sample, DI_6hvs24h)
wilcox.test(PA_DI_ratio_total ~ sample, DI_6hvs24h) 

wilcox.test(PB2_DI_ratio_total ~ sample, DI_6hvs24h, alternative="less")
wilcox.test(PB1_DI_ratio_total ~ sample, DI_6hvs24h, alternative="less")
wilcox.test(PA_DI_ratio_total ~ sample, DI_6hvs24h, alternative="less") 

# 12h vs 24h
DI_12hvs24h <- A549_DI[A549_DI$sample == "12hpi" | A549_DI$sample == "24hpi", ]
wilcox.test(PB2_DI_ratio_total ~ sample, DI_12hvs24h)
wilcox.test(PB1_DI_ratio_total ~ sample, DI_12hvs24h)
wilcox.test(PA_DI_ratio_total ~ sample, DI_12hvs24h) 

wilcox.test(PB2_DI_ratio_total ~ sample, DI_12hvs24h, alternative="less")
wilcox.test(PB1_DI_ratio_total ~ sample, DI_12hvs24h, alternative="less")
wilcox.test(PA_DI_ratio_total ~ sample, DI_12hvs24h, alternative="less") 


## DI normalized to the UMI counts at the 3' peak region
## Prepare and plot
A549_DI <- rbind(A549_6h_DI[, c(1, 14:16)], A549_12h_DI[, c(1, 14:16)], 
                 A549_24h_DI[, c(1, 14:16)])
A549_DI$sample <- rep(c("6hpi", "12hpi", "24hpi"), 
                      c(nrow(A549_6h_DI), nrow(A549_12h_DI), nrow(A549_24h_DI)))
A549_DI$sample <- factor(A549_DI$sample, levels=c("6hpi", "12hpi", "24hpi"))

A549_DI_melt <- melt(A549_DI, id.vars=c("cell", "sample"))
A549_DI_melt$variable <- gsub("_DI_ratio_3p", "", A549_DI_melt$variable)
A549_DI_melt$variable <- factor(A549_DI_melt$variable, levels=c("PB2", "PB1", "PA"))

# raincloud plot
ggplot(data=A549_DI_melt, aes(x=sample, y=value, fill=sample)) +
  facet_wrap(~variable) +
  geom_flat_violin(position=position_nudge(x=.2, y=0), alpha=.8) +
  geom_point(aes(y=value, color=sample), position=position_jitter(width=.15), size=.5, alpha=0.8) +
  geom_boxplot(width=.2, guides=FALSE, outlier.shape=NA, alpha=0, lwd=0.8) +
  scale_y_continuous(breaks=seq(0, 4, 0.5), limits=c(-0.01, 4)) +
  scale_color_manual(values=c("#b1df01", "olivedrab4", "#466b5d")) +
  scale_fill_manual(values=c("#b1df01", "olivedrab4", "#466b5d")) +
  theme(plot.title=element_text(size=15, hjust = 0.5), 
        axis.title=element_text(size=15),
        text=element_text(size=15),
        panel.background=element_rect(fill="white", color="grey50"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  labs(list(title="Distribution of the ratio of gapped viral transcripts", 
            x="Time point", y="Ratio of gapped viral transcripts"))


## Test if the distribution is statistical different
kruskal.test(PB2_DI_ratio_3p ~ sample, A549_DI)
kruskal.test(PB1_DI_ratio_3p ~ sample, A549_DI)
kruskal.test(PA_DI_ratio_3p ~ sample, A549_DI) 

# 6h vs 12h
DI_6hvs12h <- A549_DI[A549_DI$sample == "6hpi" | A549_DI$sample == "12hpi", ]
wilcox.test(PB2_DI_ratio_3p ~ sample, DI_6hvs12h)
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_6hvs12h)
wilcox.test(PA_DI_ratio_3p ~ sample, DI_6hvs12h) 

wilcox.test(PB2_DI_ratio_3p ~ sample, DI_6hvs12h, alternative="less") 
wilcox.test(PA_DI_ratio_3p ~ sample, DI_6hvs12h, alternative="greater")  

# 6h vs 24h
DI_6hvs24h <- A549_DI[A549_DI$sample == "6hpi" | A549_DI$sample == "24hpi", ]
wilcox.test(PB2_DI_ratio_3p ~ sample, DI_6hvs24h)
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_6hvs24h)
wilcox.test(PA_DI_ratio_3p ~ sample, DI_6hvs24h) 

wilcox.test(PB2_DI_ratio_3p ~ sample, DI_6hvs24h, alternative="less")
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_6hvs24h, alternative="less")
wilcox.test(PA_DI_ratio_3p ~ sample, DI_6hvs24h, alternative="less") 

# 12h vs 24h
DI_12hvs24h <- A549_DI[A549_DI$sample == "12hpi" | A549_DI$sample == "24hpi", ]
wilcox.test(PB2_DI_ratio_3p ~ sample, DI_12hvs24h)
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_12hvs24h)
wilcox.test(PA_DI_ratio_3p ~ sample, DI_12hvs24h) 

wilcox.test(PB2_DI_ratio_3p ~ sample, DI_12hvs24h, alternative="less")
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_12hvs24h, alternative="less")
wilcox.test(PA_DI_ratio_3p ~ sample, DI_12hvs24h, alternative="less") 



### HBEpC 6hpi ###
## Load data
DI_count <- read.csv("HBEpC_6h/HBEpC_6h_polymerases_DI_filtered-grouped_counts.csv",
                     row.names=1)
peak_count <- read.csv("HBEpC_6h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                       row.names=1)
peak_count <- as.data.frame(t(peak_count))
colnames(peak_count) <- c("PB2", "PB1", "PA")
load("CellQC/HBEpC/HBEpCALLsamples_QCed_gene-expression-matrix_s6_HBEpC-6h.RData")
seg_count <- s6[(nrow(s6)-7):(nrow(s6)-5), ]
seg_count <- as.data.frame(t(seg_count))

peak_count$cell <- rownames(peak_count)
seg_count$cell <- rownames(seg_count)
DI_info_master <- merge(DI_count, seg_count, by="cell")
DI_info_master <- merge(DI_info_master, peak_count, by="cell")
colnames(DI_info_master)[5:10] <- c("PB2_seg_count", "PB1_seg_count", "PA_seg_count",
                                    "PB2_peak_count", "PB1_peak_count", "PA_peak_count")

## Calculate DI ratio 
DI_info_master$PB2_DI_ratio_total <- DI_info_master$PB2_DI_count / DI_info_master$PB2_seg_count
DI_info_master$PB1_DI_ratio_total <- DI_info_master$PB1_DI_count / DI_info_master$PB1_seg_count
DI_info_master$PA_DI_ratio_total <- DI_info_master$PA_DI_count / DI_info_master$PA_seg_count

DI_info_master$PB2_DI_ratio_3p <- DI_info_master$PB2_DI_count / DI_info_master$PB2_peak_count
DI_info_master$PB1_DI_ratio_3p <- DI_info_master$PB1_DI_count / DI_info_master$PB1_peak_count
DI_info_master$PA_DI_ratio_3p <- DI_info_master$PA_DI_count / DI_info_master$PA_peak_count

# Export new table 
write.csv(DI_info_master, file="HBEpC_6h/HBEpC_6h_polymerases_DI-ratio.csv")


## Adjust DI ratios for HBEpC 6hpi specifically to eliminate the weird DI ratios due to low coverage
# For total transcripts # normalization, reset the value for cells with all the viral reads derived from gapped reads to 0
DI_info_master$PB2_DI_cov_diff_total <- DI_info_master$PB2_DI_count - DI_info_master$PB2_seg_count
DI_info_master$PB1_DI_cov_diff_total <- DI_info_master$PB1_DI_count - DI_info_master$PB1_seg_count
DI_info_master$PA_DI_cov_diff_total <- DI_info_master$PA_DI_count - DI_info_master$PA_seg_count

# Get the IDs of cells if the DI_cov_diff >= 0 
PB2_low_cov_cID_total <- as.character(DI_info_master[DI_info_master$PB2_DI_cov_diff_total >= 0, "cell"])
PB1_low_cov_cID_total <- as.character(DI_info_master[DI_info_master$PB1_DI_cov_diff_total >= 0, "cell"])
PA_low_cov_cID_total <- as.character(DI_info_master[DI_info_master$PA_DI_cov_diff_total >= 0, "cell"])

# Reset the DI ratio for the corresponding segmens in above cells to 0
DI_info_master$adj_PB2_DI_ratio_total <- DI_info_master$PB2_DI_ratio_total
DI_info_master$adj_PB1_DI_ratio_total <- DI_info_master$PB1_DI_ratio_total
DI_info_master$adj_PA_DI_ratio_total <- DI_info_master$PA_DI_ratio_total

for(i in PB2_low_cov_cID_total) {
  DI_info_master[DI_info_master$cell == i, "adj_PB2_DI_ratio_total"] <- 0
}
for(i in PB1_low_cov_cID_total) {
  DI_info_master[DI_info_master$cell == i, "adj_PB1_DI_ratio_total"] <- 0
}
for(i in PA_low_cov_cID_total) {
  DI_info_master[DI_info_master$cell == i, "adj_PA_DI_ratio_total"] <- 0
}


# For 3 peak full length reads normalization, reset the DI ratio to 0 for cells with no peak range read counts (due to a skewed coverage with peak not at 3' end)
DI_info_master$adj_PB2_DI_ratio_3p <- DI_info_master$PB2_DI_ratio_3p
DI_info_master$adj_PB1_DI_ratio_3p <- DI_info_master$PB1_DI_ratio_3p
DI_info_master$adj_PA_DI_ratio_3p <- DI_info_master$PA_DI_ratio_3p

PB2_low_cov_cID_3p <- as.character(DI_info_master[DI_info_master$PB2_peak_count == 0, "cell"])
PB1_low_cov_cID_3p <- as.character(DI_info_master[DI_info_master$PB1_peak_count == 0, "cell"])
PA_low_cov_cID_3p <- as.character(DI_info_master[DI_info_master$PA_peak_count == 0, "cell"])

for(i in PB2_low_cov_cID_3p) {
  DI_info_master[DI_info_master$cell == i, "adj_PB2_DI_ratio_3p"] <- 0
}
for(i in PB1_low_cov_cID_3p) {
  DI_info_master[DI_info_master$cell == i, "adj_PB1_DI_ratio_3p"] <- 0
}
for(i in PA_low_cov_cID_3p) {
  DI_info_master[DI_info_master$cell == i, "adj_PA_DI_ratio_3p"] <- 0
}


# Export new table 
write.csv(DI_info_master, file="HBEpC_6h/HBEpC_6h_polymerases_DI-ratio_adj.csv")


## Check the distribution of DI ratios again
DI_ratio_total <- DI_info_master[, c(1, 20:22)]
DI_ratio_total_melt <- melt(DI_ratio_total, id.vars="cell")
DI_ratio_total_melt$variable <- gsub("_DI_ratio_total", "", DI_ratio_total_melt$variable)
DI_ratio_total_melt$variable <- gsub("adj_", "", DI_ratio_total_melt$variable)
DI_ratio_total_melt$variable <- factor(DI_ratio_total_melt$variable, 
                                       levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_total_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 1, 0.1)) + 
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")

DI_ratio_3p <- DI_info_master[, c(1, 23:25)]
DI_ratio_3p_melt <- melt(DI_ratio_3p, id.vars="cell")
DI_ratio_3p_melt$variable <- gsub("_DI_ratio_3p", "", DI_ratio_3p_melt$variable)
DI_ratio_3p_melt$variable <- gsub("adj_", "", DI_ratio_3p_melt$variable)
DI_ratio_3p_melt$variable <- factor(DI_ratio_3p_melt$variable, levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_3p_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 4, 0.5)) +
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")



### HBEpC 12hpi ###
## Load data
DI_count <- read.csv("HBEpC_12h/HBEpC_12h_polymerases_DI_filtered-grouped_counts.csv",
                     row.names=1)
peak_count <- read.csv("HBEpC_12h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                       row.names=1)
peak_count <- as.data.frame(t(peak_count))
colnames(peak_count) <- c("PB2", "PB1", "PA")
load("CellQC/HBEpC/HBEpCALLsamples_QCed_gene-expression-matrix_s7_HBEpC-12h.RData")
seg_count <- s7[(nrow(s7)-7):(nrow(s7)-5), ]
seg_count <- as.data.frame(t(seg_count))

peak_count$cell <- rownames(peak_count)
seg_count$cell <- rownames(seg_count)
DI_info_master <- merge(DI_count, seg_count, by="cell")
DI_info_master <- merge(DI_info_master, peak_count, by="cell")
colnames(DI_info_master)[5:10] <- c("PB2_seg_count", "PB1_seg_count", "PA_seg_count",
                                    "PB2_peak_count", "PB1_peak_count", "PA_peak_count")

## Calculate DI ratio 
DI_info_master$PB2_DI_ratio_total <- DI_info_master$PB2_DI_count / DI_info_master$PB2_seg_count
DI_info_master$PB1_DI_ratio_total <- DI_info_master$PB1_DI_count / DI_info_master$PB1_seg_count
DI_info_master$PA_DI_ratio_total <- DI_info_master$PA_DI_count / DI_info_master$PA_seg_count

DI_info_master$PB2_DI_ratio_3p <- DI_info_master$PB2_DI_count / DI_info_master$PB2_peak_count
DI_info_master$PB1_DI_ratio_3p <- DI_info_master$PB1_DI_count / DI_info_master$PB1_peak_count
DI_info_master$PA_DI_ratio_3p <- DI_info_master$PA_DI_count / DI_info_master$PA_peak_count

# Export 
write.csv(DI_info_master, file="HBEpC_12h/HBEpC_12h_polymerases_DI-ratio.csv")


## Check the distribution of DI ratios
DI_ratio_total <- DI_info_master[, c(1, 11:13)]
DI_ratio_total_melt <- melt(DI_ratio_total, id.vars="cell")
DI_ratio_total_melt$variable <- gsub("_DI_ratio_total", "", DI_ratio_total_melt$variable)
DI_ratio_total_melt$variable <- factor(DI_ratio_total_melt$variable, 
                                       levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_total_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 1, 0.1)) + 
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")

DI_ratio_3p <- DI_info_master[, c(1, 14:16)]
DI_ratio_3p_melt <- melt(DI_ratio_3p, id.vars="cell")
DI_ratio_3p_melt$variable <- gsub("_DI_ratio_3p", "", DI_ratio_3p_melt$variable)
DI_ratio_3p_melt$variable <- factor(DI_ratio_3p_melt$variable, levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_3p_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 4, 0.5)) +
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")



### HBEpC 24hpi ###
## Load data
DI_count <- read.csv("HBEpC_24h/HBEpC_24h_polymerases_DI_filtered-grouped_counts.csv",
                     row.names=1)
peak_count <- read.csv("HBEpC_24h/reads_mapped-to-3p-peak-each-segment_per-sample_full-length_UMIcounts.csv",
                       row.names=1)
peak_count <- as.data.frame(t(peak_count))
colnames(peak_count) <- c("PB2", "PB1", "PA")
load("CellQC/HBEpC/HBEpCALLsamples_QCed_gene-expression-matrix_s8_HBEpC-24h.RData")
seg_count <- s8[((nrow(s8)-7):(nrow(s8)-5)), ]
seg_count <- as.data.frame(t(seg_count))

peak_count$cell <- rownames(peak_count)
seg_count$cell <- rownames(seg_count)
DI_info_master <- merge(DI_count, seg_count, by="cell")
DI_info_master <- merge(DI_info_master, peak_count, by="cell")
colnames(DI_info_master)[5:10] <- c("PB2_seg_count", "PB1_seg_count", "PA_seg_count",
                                    "PB2_peak_count", "PB1_peak_count", "PA_peak_count")

## Calculate DI ratio 
DI_info_master$PB2_DI_ratio_total <- DI_info_master$PB2_DI_count / DI_info_master$PB2_seg_count
DI_info_master$PB1_DI_ratio_total <- DI_info_master$PB1_DI_count / DI_info_master$PB1_seg_count
DI_info_master$PA_DI_ratio_total <- DI_info_master$PA_DI_count / DI_info_master$PA_seg_count

DI_info_master$PB2_DI_ratio_3p <- DI_info_master$PB2_DI_count / DI_info_master$PB2_peak_count
DI_info_master$PB1_DI_ratio_3p <- DI_info_master$PB1_DI_count / DI_info_master$PB1_peak_count
DI_info_master$PA_DI_ratio_3p <- DI_info_master$PA_DI_count / DI_info_master$PA_peak_count

# Export 
write.csv(DI_info_master, file="HBEpC_24h/HBEpC_24h_polymerases_DI-ratio.csv")


## Check the distribution of DI ratios
DI_ratio_total <- DI_info_master[, c(1, 11:13)]
DI_ratio_total_melt <- melt(DI_ratio_total, id.vars="cell")
DI_ratio_total_melt$variable <- gsub("_DI_ratio_total", "", DI_ratio_total_melt$variable)
DI_ratio_total_melt$variable <- factor(DI_ratio_total_melt$variable, 
                                       levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_total_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 1, 0.1)) + 
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")

DI_ratio_3p <- DI_info_master[, c(1, 14:16)]
DI_ratio_3p_melt <- melt(DI_ratio_3p, id.vars="cell")
DI_ratio_3p_melt$variable <- gsub("_DI_ratio_3p", "", DI_ratio_3p_melt$variable)
DI_ratio_3p_melt$variable <- factor(DI_ratio_3p_melt$variable, levels=c("PB2", "PB1", "PA"))
ggplot(DI_ratio_3p_melt, aes(x=variable, y=value, fill=variable)) + 
  geom_violin() +
  geom_jitter(size=0.3) +
  scale_y_continuous(breaks=seq(0, 4, 0.5)) +
  theme(text=element_text(size=15),
        panel.background=element_rect(fill="white"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  scale_fill_manual(values=c("slateblue", "gold", "skyblue")) +
  labs(x="Segments", y="DI ratio")



### All three HBEpC time points together ###
## Load data
HBEpC_6h_DI <- read.csv("HBEpC_6h/HBEpC_6h_polymerases_DI-ratio_adj.csv", row.names=1)
HBEpC_12h_DI <- read.csv("HBEpC_12h/HBEpC_12h_polymerases_DI-ratio.csv", row.names=1)
HBEpC_24h_DI <- read.csv("HBEpC_24h/HBEpC_24h_polymerases_DI-ratio.csv", row.names=1)

## DI normalized to the UMI counts at the 3' peak region
## Prepare and plot
HBEpC_6h_DI <- HBEpC_6h_DI[, c(1, 23:25)]
colnames(HBEpC_6h_DI) <- c("cell", "PB2_DI_ratio_3p", "PB1_DI_ratio_3p", "PA_DI_ratio_3p")
HBEpC_DI <- rbind(HBEpC_6h_DI, HBEpC_12h_DI[, c(1, 14:16)], HBEpC_24h_DI[, c(1, 14:16)])
HBEpC_DI$sample <- rep(c("6hpi", "12hpi", "24hpi"), 
                       c(nrow(HBEpC_6h_DI), nrow(HBEpC_12h_DI), nrow(HBEpC_24h_DI)))
HBEpC_DI$sample <- factor(HBEpC_DI$sample, levels=c("6hpi", "12hpi", "24hpi"))

HBEpC_DI_melt <- melt(HBEpC_DI, id.vars=c("cell", "sample"))
HBEpC_DI_melt$variable <- gsub("_DI_ratio_3p", "", HBEpC_DI_melt$variable)
HBEpC_DI_melt$variable <- factor(HBEpC_DI_melt$variable, levels=c("PB2", "PB1", "PA"))

# raincloud plot
ggplot(data=HBEpC_DI_melt, aes(x=sample, y=value, fill=sample)) +
  facet_wrap(~variable) +
  geom_flat_violin(position=position_nudge(x=.2, y=0), alpha=.8) +
  geom_point(aes(y=value, color=sample), position=position_jitter(width=.15), size=.5, alpha=0.8) +
  geom_boxplot(width=.2, guides=FALSE, outlier.shape=NA, alpha=0, lwd=0.8) +
  scale_y_continuous(breaks=seq(0, 4, 0.5), limits=c(-0.01, 4)) +
  scale_color_manual(values=c("#b1df01", "olivedrab4", "#466b5d")) +
  scale_fill_manual(values=c("#b1df01", "olivedrab4", "#466b5d")) +
  theme(plot.title=element_text(size=15, hjust = 0.5), 
        axis.title=element_text(size=15),
        text=element_text(size=15),
        panel.background=element_rect(fill="white", color="grey50"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  labs(list(title="Distribution of relative abundance of viral transcripts", 
            x="Time point", y="Relative abundance of viral transcripts"))


## Test if the distribution is statistical different
kruskal.test(PB2_DI_ratio_3p ~ sample, HBEpC_DI)
kruskal.test(PB1_DI_ratio_3p ~ sample, HBEpC_DI)
kruskal.test(PA_DI_ratio_3p ~ sample, HBEpC_DI) 

# 6h vs 12h
DI_6hvs12h <- HBEpC_DI[HBEpC_DI$sample == "6hpi" | HBEpC_DI$sample == "12hpi", ]
wilcox.test(PB2_DI_ratio_3p ~ sample, DI_6hvs12h)
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_6hvs12h)
wilcox.test(PA_DI_ratio_3p ~ sample, DI_6hvs12h) 

wilcox.test(PB2_DI_ratio_3p ~ sample, DI_6hvs12h, alternative="less") 
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_6hvs12h, alternative="less") 

# 6h vs 24h
DI_6hvs24h <- HBEpC_DI[HBEpC_DI$sample == "6hpi" | HBEpC_DI$sample == "24hpi", ]
wilcox.test(PB2_DI_ratio_3p ~ sample, DI_6hvs24h)
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_6hvs24h)
wilcox.test(PA_DI_ratio_3p ~ sample, DI_6hvs24h) 

wilcox.test(PB2_DI_ratio_3p ~ sample, DI_6hvs24h, alternative="less")
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_6hvs24h, alternative="less")
wilcox.test(PA_DI_ratio_3p ~ sample, DI_6hvs24h, alternative="less") 

# 12h vs 24h
DI_12hvs24h <- HBEpC_DI[HBEpC_DI$sample == "12hpi" | HBEpC_DI$sample == "24hpi", ]
wilcox.test(PB2_DI_ratio_3p ~ sample, DI_12hvs24h)
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_12hvs24h)
wilcox.test(PA_DI_ratio_3p ~ sample, DI_12hvs24h) 

wilcox.test(PB2_DI_ratio_3p ~ sample, DI_12hvs24h, alternative="less")
wilcox.test(PB1_DI_ratio_3p ~ sample, DI_12hvs24h, alternative="less")
wilcox.test(PA_DI_ratio_3p ~ sample, DI_12hvs24h, alternative="less") 


## DI normalized to the total UMI counts for each segment
## Prepare and plot
HBEpC_6h_DI <- read.csv("HBEpC_6h/HBEpC_6h_polymerases_DI-ratio_adj.csv", row.names=1)
HBEpC_6h_DI <- HBEpC_6h_DI[, c(1, 20:22)]
colnames(HBEpC_6h_DI) <- c("cell", "PB2_DI_ratio_total", "PB1_DI_ratio_total", "PA_DI_ratio_total")
HBEpC_DI <- rbind(HBEpC_6h_DI, HBEpC_12h_DI[, c(1, 11:13)], HBEpC_24h_DI[, c(1, 11:13)])
HBEpC_DI$sample <- rep(c("6hpi", "12hpi", "24hpi"), 
                       c(nrow(HBEpC_6h_DI), nrow(HBEpC_12h_DI), nrow(HBEpC_24h_DI)))
HBEpC_DI$sample <- factor(HBEpC_DI$sample, levels=c("6hpi", "12hpi", "24hpi"))

HBEpC_DI_melt <- melt(HBEpC_DI, id.vars=c("cell", "sample"))
HBEpC_DI_melt$variable <- gsub("_DI_ratio_total", "", HBEpC_DI_melt$variable)
HBEpC_DI_melt$variable <- factor(HBEpC_DI_melt$variable, levels=c("PB2", "PB1", "PA"))

# raincloud plot
HBEpC_DI_melt$value <- HBEpC_DI_melt$value * 100
ggplot(data=HBEpC_DI_melt, aes(x=sample, y=value, fill=sample)) +
  facet_wrap(~variable) +
  geom_flat_violin(position=position_nudge(x=.2, y=0), alpha=.8) +
  geom_point(aes(y=value, color=sample), position=position_jitter(width=.15), size=.5, alpha=0.8) +
  geom_boxplot(width=.2, guides=FALSE, outlier.shape=NA, alpha=0, lwd=0.8) +
  scale_y_continuous(breaks=seq(0, 80, 10), limits=c(-0.01, 85)) +
  scale_color_manual(values=c("#b1df01", "olivedrab4", "#466b5d")) +
  scale_fill_manual(values=c("#b1df01", "olivedrab4", "#466b5d")) +
  theme(plot.title=element_text(size=15, hjust = 0.5), 
        axis.title=element_text(size=15),
        text=element_text(size=15),
        panel.background=element_rect(fill="white", color="grey50"),
        panel.grid.major=element_line(color="grey90"),
        legend.position="none") +
  labs(list(title="Distribution of relative abundance of viral transcripts", 
            x="Time point", y="Relative abundance of viral transcripts"))


## Test if the distribution is statistical different
kruskal.test(PB2_DI_ratio_total ~ sample, HBEpC_DI)
kruskal.test(PB1_DI_ratio_total ~ sample, HBEpC_DI)
kruskal.test(PA_DI_ratio_total ~ sample, HBEpC_DI) 

# 6h vs 12h
DI_6hvs12h <- HBEpC_DI[HBEpC_DI$sample == "6hpi" | HBEpC_DI$sample == "12hpi", ]
wilcox.test(PB2_DI_ratio_total ~ sample, DI_6hvs12h)
wilcox.test(PB1_DI_ratio_total ~ sample, DI_6hvs12h)
wilcox.test(PA_DI_ratio_total ~ sample, DI_6hvs12h) 

wilcox.test(PB2_DI_ratio_total ~ sample, DI_6hvs12h, alternative="less")  
wilcox.test(PB1_DI_ratio_total ~ sample, DI_6hvs12h, alternative="less")  

# 6h vs 24h
DI_6hvs24h <- HBEpC_DI[HBEpC_DI$sample == "6hpi" | HBEpC_DI$sample == "24hpi", ]
wilcox.test(PB2_DI_ratio_total ~ sample, DI_6hvs24h)
wilcox.test(PB1_DI_ratio_total ~ sample, DI_6hvs24h)
wilcox.test(PA_DI_ratio_total ~ sample, DI_6hvs24h) 

wilcox.test(PB2_DI_ratio_total ~ sample, DI_6hvs24h, alternative="less") 
wilcox.test(PB1_DI_ratio_total ~ sample, DI_6hvs24h, alternative="less") 
wilcox.test(PA_DI_ratio_total ~ sample, DI_6hvs24h, alternative="less")  

# 12h vs 24h
DI_12hvs24h <- HBEpC_DI[HBEpC_DI$sample == "12hpi" | HBEpC_DI$sample == "24hpi", ]
wilcox.test(PB2_DI_ratio_total ~ sample, DI_12hvs24h)
wilcox.test(PB1_DI_ratio_total ~ sample, DI_12hvs24h)
wilcox.test(PA_DI_ratio_total ~ sample, DI_12hvs24h) 

wilcox.test(PB2_DI_ratio_total ~ sample, DI_12hvs24h, alternative="less") 
wilcox.test(PB1_DI_ratio_total ~ sample, DI_12hvs24h, alternative="less") 
wilcox.test(PA_DI_ratio_total ~ sample, DI_12hvs24h, alternative="less") 

```


